---
title: "MYK065"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r Library}
library(tidyverse)
library(Seurat)
library(monocle)
library(ggpubr)
library(RColorBrewer)
library(readxl)
library(biomaRt)
#BiocManager::install("monocle")
library(monocle)
#source("~/Documents/MCW/3_PhD/Bioinformatics_Collaborations/Functions_themes.R")
```

**Combine D9 (MYK048) and D3 (MYK065) data with naive (D0) published lung data**
Naive data from Angelidis et al., 2019, Nat Comm (10.1038/s41467-019-08831-9)
For naive data, RData file needs to be split into young and old based on provided metadata

```{r Load data}
### Note for D3 data - Renaming mice for convenience
# Mice called Y2 and O2 in the experiment will now be Y1 and O1; Y3 and O3 will be Y2 and O2

D3_Y1_raw <- Read10X("Matrix_files/Young_2/filtered_feature_bc_matrix/")
D3_Y2_raw <- Read10X("Matrix_files/Young_3/filtered_feature_bc_matrix/")
D3_Y3_raw <- Read10X("../../AKB048_scRNAseq_Dec2022/Analysis/Matrix_files/Young_D3/filtered_feature_bc_matrix/")

D3_O1_raw <- Read10X("Matrix_files/Old_2/filtered_feature_bc_matrix/")
D3_O2_raw <- Read10X("Matrix_files/Old_3/filtered_feature_bc_matrix/")
D3_O3_raw <- Read10X("../../AKB048_scRNAseq_Dec2022/Analysis/Matrix_files/Old_D3/filtered_feature_bc_matrix/")

D9_Y1_raw <- Read10X("../../MYK048_ILC_scRNAseq/Analysis/Matrix_files/Y1/filtered_feature_bc_matrix/")
D9_Y2_raw <- Read10X("../../MYK048_ILC_scRNAseq/Analysis/Matrix_files/Y2/filtered_feature_bc_matrix/")
D9_Y3_raw <- Read10X("../../AKB048_scRNAseq_Dec2022/Analysis/Matrix_files/Young_D9/filtered_feature_bc_matrix/")

D9_O1_raw <- Read10X("../../MYK048_ILC_scRNAseq/Analysis/Matrix_files/O1/filtered_feature_bc_matrix/")
D9_O2_raw <- Read10X("../../MYK048_ILC_scRNAseq/Analysis/Matrix_files/O2/filtered_feature_bc_matrix/")
D9_O3_raw <- Read10X("../../AKB048_scRNAseq_Dec2022/Analysis/Matrix_files/Old_D9/filtered_feature_bc_matrix/")


### This data contains both GEX and HTO (hashtag oligo) data
# 10X data contains more than one type and is being returned as a list of sparce matrices, one for each data modality (GEX and HTO) per sample

# Change HTO rownames so they match
rownames(D3_Y1_raw$`Antibody Capture`) <- "IV-CD45"
rownames(D3_Y2_raw$`Antibody Capture`) <- "IV-CD45"

rownames(D3_O1_raw$`Antibody Capture`) <- "IV-CD45"
rownames(D3_O2_raw$`Antibody Capture`) <- "IV-CD45"

rownames(D9_Y1_raw$`Antibody Capture`) <- "IV-CD45"
rownames(D9_Y2_raw$`Antibody Capture`) <- "IV-CD45"

rownames(D9_O1_raw$`Antibody Capture`) <- "IV-CD45"
rownames(D9_O2_raw$`Antibody Capture`) <- "IV-CD45"

dim(D3_Y1_raw$`Gene Expression`) # 32285  3600
dim(D3_Y2_raw$`Gene Expression`) # 32285  4125
dim(D3_Y3_raw) # 32285  8555

dim(D3_O1_raw$`Gene Expression`) # 32285  7133
dim(D3_O2_raw$`Gene Expression`) # 32285  7232
dim(D3_O3_raw) # 32285  6798

dim(D9_Y1_raw$`Gene Expression`) # 32285  6104
dim(D9_Y2_raw$`Gene Expression`) # 32285  7071
dim(D9_Y3_raw) # 32285  12189

dim(D9_O1_raw$`Gene Expression`) # 32285  3647
dim(D9_O2_raw$`Gene Expression`) # 32285  8248
dim(D9_O3_raw) # 32285  7802

```

```{r Load public naive lung data}
load("Public_naive_data/GSE124872_raw_counts_single_cell.RData")
D0_Y <- raw_counts[, grep("young", colnames(raw_counts))]
D0_O <- raw_counts[, grep("old", colnames(raw_counts))]
rm(raw_counts)

dim(D0_Y) # 21969  7672
dim(D0_O) # 21969  7141

# Remove young/old prefixes from cell IDs
colnames(D0_Y) <- substring(colnames(D0_Y), 9)
colnames(D0_O) <- substring(colnames(D0_O), 7)
```

```{r Create Seurat objects}
D0_Y <- CreateSeuratObject(D0_Y, project = "D0_Y")
D0_O <- CreateSeuratObject(D0_O, project = "D0_O")

D3_Y1 <- CreateSeuratObject(D3_Y1_raw$`Gene Expression`, project = "D3_Y1")
D3_Y2 <- CreateSeuratObject(D3_Y2_raw$`Gene Expression`, project = "D3_Y2")
D3_Y3 <- CreateSeuratObject(D3_Y3_raw, project = "D3_Y3")

D3_O1 <- CreateSeuratObject(D3_O1_raw$`Gene Expression`, project = "D3_O1")
D3_O2 <- CreateSeuratObject(D3_O2_raw$`Gene Expression`, project = "D3_O2")
D3_O3 <- CreateSeuratObject(D3_O3_raw, project = "D3_O3")


D9_Y1 <- CreateSeuratObject(D9_Y1_raw$`Gene Expression`, project = "D9_Y1")
D9_Y2 <- CreateSeuratObject(D9_Y2_raw$`Gene Expression`, project = "D9_Y2")
D9_Y3 <- CreateSeuratObject(D9_Y3_raw, project = "D9_Y3")

D9_O1 <- CreateSeuratObject(D9_O1_raw$`Gene Expression`, project = "D9_O1")
D9_O2 <- CreateSeuratObject(D9_O2_raw$`Gene Expression`, project = "D9_O2")
D9_O3 <- CreateSeuratObject(D9_O3_raw, project = "D9_O3")


D3_Y1[["HTO"]] <- CreateAssayObject(D3_Y1_raw$`Antibody Capture`)
D3_O1[["HTO"]] <- CreateAssayObject(D3_O1_raw$`Antibody Capture`)
D3_Y2[["HTO"]] <- CreateAssayObject(D3_Y2_raw$`Antibody Capture`)
D3_O2[["HTO"]] <- CreateAssayObject(D3_O2_raw$`Antibody Capture`)
D9_Y1[["HTO"]] <- CreateAssayObject(D9_Y1_raw$`Antibody Capture`)
D9_O1[["HTO"]] <- CreateAssayObject(D9_O1_raw$`Antibody Capture`)
D9_Y2[["HTO"]] <- CreateAssayObject(D9_Y2_raw$`Antibody Capture`)
D9_O2[["HTO"]] <- CreateAssayObject(D9_O2_raw$`Antibody Capture`)


D0_Y$Age <- "Young"
D0_Y$Timepoint <- "Day 0"
D0_Y <- RenameCells(D0_Y, add.cell.id = "D0_Y")

D0_O$Age <- "Aged"
D0_O$Timepoint <- "Day 0"
D0_O <- RenameCells(D0_O, add.cell.id = "D0_O")

D3_Y1$Age <- "Young"
D3_Y1$Timepoint <- "Day 3"
D3_Y1 <- RenameCells(D3_Y1, add.cell.id = "D3_Y1")

D3_Y2$Age <- "Young"
D3_Y2$Timepoint <- "Day 3"
D3_Y2 <- RenameCells(D3_Y2, add.cell.id = "D3_Y2")

D3_Y3$Age <- "Young"
D3_Y3$Timepoint <- "Day 3"
D3_Y3 <- RenameCells(D3_Y3, add.cell.id = "D3_Y3")

D3_O1$Age <- "Aged"
D3_O1$Timepoint <- "Day 3"
D3_O1 <- RenameCells(D3_O1, add.cell.id = "D3_O1")

D3_O2$Age <- "Aged"
D3_O2$Timepoint <- "Day 3"
D3_O2 <- RenameCells(D3_O2, add.cell.id = "D3_O2")

D3_O3$Age <- "Aged"
D3_O3$Timepoint <- "Day 3"
D3_O3 <- RenameCells(D3_O3, add.cell.id = "D3_O3")

D9_Y1$Age <- "Young"
D9_Y1$Timepoint <- "Day 9"
D9_Y1 <- RenameCells(D9_Y1, add.cell.id = "D9_Y1")

D9_Y2$Age <- "Young"
D9_Y2$Timepoint <- "Day 9"
D9_Y2 <- RenameCells(D9_Y2, add.cell.id = "D9_Y2")

D9_Y3$Age <- "Young"
D9_Y3$Timepoint <- "Day 9"
D9_Y3 <- RenameCells(D9_Y3, add.cell.id = "D9_Y3")

D9_O1$Age <- "Aged"
D9_O1$Timepoint <- "Day 9"
D9_O1 <- RenameCells(D9_O1, add.cell.id = "D9_O1")

D9_O2$Age <- "Aged"
D9_O2$Timepoint <- "Day 9"
D9_O2 <- RenameCells(D9_O2, add.cell.id = "D9_O2")

D9_O3$Age <- "Aged"
D9_O3$Timepoint <- "Day 9"
D9_O3 <- RenameCells(D9_O3, add.cell.id = "D9_O3")

rm(list = grep("_raw", ls(), value = T))
```

```{r Quality control}
D0_Y$percent.mt <- PercentageFeatureSet(D0_Y, pattern = "^mt-")
VlnPlot(D0_Y, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D0_Y.png"), units="in", width = 6, height = 5, dpi=1200)
D0_Y <- subset(D0_Y, subset = nFeature_RNA > 200 & nFeature_RNA < 1000 & percent.mt < 10)

D0_O$percent.mt <- PercentageFeatureSet(D0_O, pattern = "^mt-")
VlnPlot(D0_O, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D0_O.png"), units="in", width = 6, height = 5, dpi=1200)
D0_O <- subset(D0_O, subset = nFeature_RNA > 200 & nFeature_RNA < 1000 & percent.mt < 10)

D3_Y1$percent.mt <- PercentageFeatureSet(D3_Y1, pattern = "^mt-")
VlnPlot(D3_Y1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D3_Y1.png"), units="in", width = 6, height = 5, dpi=1200)
D3_Y1 <- subset(D3_Y1, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D3_Y2$percent.mt <- PercentageFeatureSet(D3_Y2, pattern = "^mt-")
VlnPlot(D3_Y2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D3_Y2.png"), units="in", width = 6, height = 5, dpi=1200)
D3_Y2 <- subset(D3_Y2, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D3_Y3$percent.mt <- PercentageFeatureSet(D3_Y3, pattern = "^mt-")
VlnPlot(D3_Y3, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D3_Y3.png"), units="in", width = 6, height = 5, dpi=1200)
D3_Y3 <- subset(D3_Y3, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D3_O1$percent.mt <- PercentageFeatureSet(D3_O1, pattern = "^mt-")
VlnPlot(D3_O1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D3_O1.png"), units="in", width = 6, height = 5, dpi=1200)
D3_O1 <- subset(D3_O1, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D3_O2$percent.mt <- PercentageFeatureSet(D3_O2, pattern = "^mt-")
VlnPlot(D3_O2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D3_O2.png"), units="in", width = 6, height = 5, dpi=1200)
D3_O2 <- subset(D3_O2, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D3_O3$percent.mt <- PercentageFeatureSet(D3_O3, pattern = "^mt-")
VlnPlot(D3_O3, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D3_O3.png"), units="in", width = 6, height = 5, dpi=1200)
D3_O3 <- subset(D3_O3, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D9_Y1$percent.mt <- PercentageFeatureSet(D9_Y1, pattern = "^mt-")
VlnPlot(D9_Y1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D9_Y1.png"), units="in", width = 6, height = 5, dpi=1200)
D9_Y1 <- subset(D9_Y1, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D9_Y2$percent.mt <- PercentageFeatureSet(D9_Y2, pattern = "^mt-")
VlnPlot(D9_Y2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D9_Y2.png"), units="in", width = 6, height = 5, dpi=1200)
D9_Y2 <- subset(D9_Y2, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D9_Y3$percent.mt <- PercentageFeatureSet(D9_Y3, pattern = "^mt-")
VlnPlot(D9_Y3, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D9_Y3.png"), units="in", width = 6, height = 5, dpi=1200)
D9_Y3 <- subset(D9_Y3, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D9_O1$percent.mt <- PercentageFeatureSet(D9_O1, pattern = "^mt-")
VlnPlot(D9_O1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D9_O1.png"), units="in", width = 6, height = 5, dpi=1200)
D9_O1 <- subset(D9_O1, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D9_O2$percent.mt <- PercentageFeatureSet(D9_O2, pattern = "^mt-")
VlnPlot(D9_O2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D9_O2.png"), units="in", width = 6, height = 5, dpi=1200)
D9_O2 <- subset(D9_O2, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

D9_O3$percent.mt <- PercentageFeatureSet(D9_O3, pattern = "^mt-")
VlnPlot(D9_O3, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ggsave(paste0(getwd(),"/ImageOutput/QC/QC_D9_O3.png"), units="in", width = 6, height = 5, dpi=1200)
D9_O3 <- subset(D9_O3, subset = nFeature_RNA > 200 & nFeature_RNA < 3500 & percent.mt < 10)

dim(D0_Y) # 32285  7227
dim(D0_O) # 32285  6502

dim(D3_Y1) # 32285  2719
dim(D3_Y2) # 32285  3048
dim(D3_Y3) # 32285  8100

dim(D3_O1) # 32285  5860
dim(D3_O2) # 32285  5096
dim(D3_O3) # 32285  6343

dim(D9_Y1) # 32285  4239
dim(D9_Y2) # 32285  5097
dim(D9_Y3) # 32285  11907

dim(D9_O1) # 32285  2620
dim(D9_O2) # 32285  6596
dim(D9_O3) # 32285  7717

```

```{r Integrate}
# Create a list of Seurat objects for integration
Flu.list <- list(D0_Y, D0_O, D3_Y1, D3_Y2, D3_Y3, D3_O1, D3_O2, D3_O3, D9_Y1, D9_Y2, D9_Y3, D9_O1, D9_O2, D9_O3)

for (i in 1:length(Flu.list)) {
  Flu.list[[i]] <- NormalizeData(Flu.list[[i]], verbose = F)
  Flu.list[[i]] <- FindVariableFeatures(Flu.list[[i]], selection.method = "vst", nfeatures = 2000, verbose = F)
}

Flu.anchors <- FindIntegrationAnchors(object.list = Flu.list, dims = 1:50)
saveRDS(Flu.anchors, "RDS_files.nosync/flu_anchors.rds")
Flu <- IntegrateData(anchorset = Flu.anchors, dims = 1:50) # npcs should be 30?
```

```{r Regress}
DefaultAssay(Flu) <- "integrated"

dim(Flu) # 83071 cells

# Cell cycle
cc.genes <- Seurat::cc.genes
cc.genes <- lapply(cc.genes, function(x) {paste0(
  substring(x, 1, 1),
  tolower(substring(x, 2))) # Change capitalization
})

Flu <- CellCycleScoring(Flu, s.features = cc.genes$s.genes, g2m.features = cc.genes$g2m.genes, set.ident = F)
Flu <- ScaleData(Flu, features = rownames(Flu), vars.to.regress = c("S.Score", "G2M.Score"))

# The following features are not present in the object: Mcm5, Pcna, Tyms, Fen1, Mcm2, Mcm4, Rrm1, Ung, Gins2, Mcm6, Cdca7, Prim1, Mlf1ip, Rfc2, Rpa2, Nasp, Rad51ap1, Gmnn, Wdr76, Slbp, Ccne2, Ubr7, Pold3, Msh2, Atad2, Rad51, Cdc45, Cdc6, Exo1, Tipin, Dscc1, Blm, Casp8ap2, Usp1, Pola1, Chaf1b, Brip1, E2f8, not searching for symbol synonyms
# The following features are not present in the object: Ndc80, Nuf2, Tmpo, Tacc3, Fam64a, Bub1, Anp32e, Gtse1, Hjurp, Hn1, Ttk, Cdc25c, Kif2c, Rangap1, Ncapd2, Dlgap5, Cdca2, Ect2, Aurka, Psrc1, Anln, Lbr, Ckap5, Ctcf, Nek2, G2e3, Gas2l3, Cbx5, not searching for symbol synonyms
```

# This is where I am running the program today

```{r UMAP}
DefaultAssay(Flu) <- "integrated"

Flu <- RunPCA(Flu, npcs = 50, verbose = F)
Flu <- RunUMAP(Flu, reduction = "pca", dims = 1:50)
Flu <- FindNeighbors(Flu, reduction = "pca", dims = 1:50)
Flu <- FindClusters(Flu, resolution = 1.2)
# Res 0.4 - 1 singletons identified. 31 final clusters. - Mouj
# Res 0.5 - 1 singletons identified. 36 final clusters. - Mouj
# Res 0.6 - 1 singletons identified. 38 final clusters. - Mouj
### Use res 0.6, separates CD4 and CD8 T cells

### Reorder factor levels
Flu$orig.ident <- factor(Flu$orig.ident, levels = c("D0_Y", "D0_O", "D3_Y1", "D3_Y2", "D3_Y3","D3_O1", "D3_O2", "D3_O3", "D9_Y1", "D9_Y2", "D9_Y3", "D9_O1", "D9_O2", "D9_O3"))
Flu$Age <- factor(Flu$Age, levels = c("Young", "Aged"))
Flu$Timepoint_Age <- paste0(Flu$Timepoint, " ", Flu$Age) %>% factor(levels = c("Day 0 Young", "Day 0 Aged", "Day 3 Young", "Day 3 Aged", "Day 9 Young", "Day 9 Aged"))

DimPlot(Flu, label = T)
# ggsave("Figures/01_UMAP.png", dpi = 300)
DimPlot(Flu, split.by = "orig.ident", ncol = 5)
# ggsave("Figures/01_UMAP_origident_split.png", dpi = 300, width = 7.29*2, height = 4.51*2)
DimPlot(Flu, split.by = "Age")
# ggsave("Figures/01_UMAP_Age_split.png", dpi = 300)
DimPlot(Flu, split.by = "Timepoint")
# ggsave("Figures/01_UMAP_Timepoint_split.png", dpi = 300, width = 7.29*1.25, height = 4.51*1.25)
DimPlot(Flu, split.by = "Timepoint_Age", ncol = 2)
# ggsave("Figures/01_UMAP_TimepointAge_split.png", dpi = 300, width = 7.29*2, height = 4.51*2)
DimPlot(Flu, group.by = "Phase")
# ggsave("Figures/01_UMAP_Phase.png", dpi = 300)
```

```{r Save Load}
saveRDS(Flu, "RDS_files.nosync/Flu_D0_D3_D9_RJB.RDS")
#Flu <- readRDS("RDS_files.nosync/Flu_D0_D3_D9_RJB.RDS")
```


```{r Cluster breakdown}
table(Flu@active.ident, Flu$orig.ident) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency")
# ggsave("Figures/02_Cluster_barplot_origident.pdf")

table(Flu@active.ident, Flu$Timepoint) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency")
# ggsave("Figures/02_Cluster_barplot_Timepoint.pdf")

table(Flu@active.ident, Flu$Age) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency")
# ggsave("Figures/02_Cluster_barplot_Age.pdf")

table(Flu@active.ident, Flu$Timepoint_Age) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency")
# ggsave("Figures/02_Cluster_barplot_TimepointAge.pdf")

table(Flu@active.ident, Flu$Timepoint_Age) %>%
  prop.table(margin = 2) * 100
```

```{r Markers}
DefaultAssay(Flu) <- "RNA"

### Not using FindAllMarkers, takes ~10 minutes per cluster (would result in ~5 hour total runtime)

# Overall markers
DotPlot(Flu, features = c("Ptprc", "Ear2", "Ly6c2", "Ly6g", "Itgam", "Adgre1", "Itgax", "Col1a2", "Scgb1a1", "Sftpd", "Rtkn2", "Cd3e", "CD4", "Cd8a", "Trdc", "Cd19", "Mzb1", "Pecam1", "Car4", "Vwf", "Klrg1", "Slc18a2", "Top2a")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

DotPlot(Flu, features = c("Cd3e", "CD4", "Cd8a"))

# T cells
FeaturePlot(Flu, features = c("CD4", "Cd8a"))
FeaturePlot(Flu, features = c("Nkg7", "Top2a", "Trdc"), order = T)

# B cells / AT1
FeaturePlot(Flu, features = c("Cd19", "Mzb1", "Igha", "Hells"), order = T)
FeaturePlot(Flu, features = c("Sftpd", "Foxj1", "Scgb1a1", "Bpifb1"), order = T)
FeaturePlot(Flu, features = c("Hba-a1"), order = T)
FeaturePlot(Flu, features = c("Hopx"), order = T)
FeaturePlot(Flu, features = c("Igha"), order = T)


# Monocytes
FeaturePlot(Flu, features = c("Ly6g", "Cd274", "Itgae", "Ccr7"), order = T)
FeaturePlot(Flu, features = c("Ear2", "Ly6c2", "C1qb", "Top2a"), order = T)
FeaturePlot(Flu, features = c("Hba-a1"), order = T)
FeaturePlot(Flu, features = c("Nkg7", "Fn1"), order = T)
FeaturePlot(Flu, features = c("Cx3cr1"), order = T)
FeaturePlot(Flu, features = c("Cd14", 'Fcgr3', "Itgam"), order = T)


# Endothelial
FeaturePlot(Flu, features = c("Pecam1", "Vwf", "Ppbp", "Car4"), order = F)
FeaturePlot(Flu, features = c("Myh10", "Col14a1", "Msln", "Fcer1a"), order = T)
FeaturePlot(Flu, features = c("Pdgfrb",'Itga2','Pf4'), order = T)
FeaturePlot(Flu, features = c("Nkg7", "Fn1"), order = T)
FeaturePlot(Flu, features = c("Prox1", "Fn1"), order = T)


# Use higher thresholds to decrease computing time
Flu_markers_C13 <- FindMarkers(Flu, ident.1 = 42, only.pos = T, min.pct = 0.25, logfc.threshold = 0.5)
# Flu_markers_C24 <- FindMarkers(Flu, ident.1 = 24, only.pos = T, min.pct = 0.25, logfc.threshold = 0.5)
# Flu_markers_C32 <- FindMarkers(Flu, ident.1 = 32, only.pos = T, min.pct = 0.25, logfc.threshold = 0.5)
# Flu_markers_C34 <- FindMarkers(Flu, ident.1 = 34, only.pos = T, min.pct = 0.25, logfc.threshold = 0.5)
# Flu_markers_C35 <- FindMarkers(Flu, ident.1 = 35, only.pos = T, min.pct = 0.25, logfc.threshold = 1)
# Flu_markers_C37 <- FindMarkers(Flu, ident.1 = 37, only.pos = T, min.pct = 0.25, logfc.threshold = 1)

Flu_markers_C13 <- FindMarkers(Flu, ident.1 = 33, only.pos = T, min.pct = 0.25, logfc.threshold = 0.5)


# C35 - Mouj # C26 - RJB
DotPlot(Flu, features = c("Ptprc", "Ccr7", "Il12b", "Ccl22", "Fscn1", "Cxcl16", "Marcks", "Fabp5", "Cd83")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

Cluster - Identity - Genes (**Res 1**) - RJB
0 - Naive B cells
1 - Endothelial cells
2 - CD8 T cells
3 - Aveolar Macrophages
4 - Intersitial macrophages
5 - Monocytes
6 - CD4 T cells
7 - Mitotic B cells
8 - Neutrophils
9 - Macrophages
10 - Club cells
11 - Plasma cells
12 - Matrix Fibtroblasts
13 - Car4+ endothelial cells
14 - NK
15 - Mitotic B cells
16 - Platelets
17 - Myofibroblasts
18 - cDC
19 - Vwf+ endothelial cells
20 - Mitotic T cells
21 - Goblet cells
22 - Mesothelial cells
23 - AT2
24 - Gamma delta T cells
25 - Myofibroblasts
26 - Cilliated cells
27 - Supressive Neutrophils
28 - Lymphatic endothelial cells
29 - RBC
30 - **JUNK**
31 - Myofibroblasts
32 - Ccr7+ DC
33 - Mitotic B cells
34 - Mitotic
35 - Pericytes
36 - Mast Cells
37 - Ccr9 - remove
38 - IghA+ B cells
39 - 

Cluster - Identity - Genes (**Res 0.8**) - RJB
0 - naive B cells
1 - Endothelial
2 - CD8 - T cells
3 - Aveolar Macrophages
4 - Plasma
5 - Monocytes
6 - CD4
7 - Neutrophils
8 - Monocytes?
9 - Cx3Cr1 Macrophages
10 - Interstitial Macrophages
11 - Matrix Fibroblasts
12 - Club Cells
13 - Myofibroblasts
14 - Platelets
15 - AT2/AT1 - AT1 is at the tip
16 - Car4+ Endothelial Cells
17 - G2M - Bcells - Hmgb2
18 - Mitotic - Top2a
19 - cDC
20 - NK cells
21 - gamma-delta T-cells - Trdc
22 - S.Phase Mitotic B-cells - Hells
23 - Goblet Cells 
24 - Vwf+ Endothelial
25 - Mesothelial Cells
26 - RBC
27 - Ciliated
28 - Lymphatic Endothelial Cells (Ccl21a)
29 - Ccr7+ DCs
30 - **Impure/doublet mix - T cells and endothelial cells - REMOVE** - Cd93, Itga1, Adgrf5, Pecam1, high nFeature_RNA
31 - Pericytes
32 - Mast Cells
33 - CCr9
34 - IgA+ B cells
35 - 
36 - 
37 - Fn1+ Macrophages



Cluster - Identity - Genes (**Res 0.6**) - Mouj
0 - Naive B cells - Cd19, CD44-lo
1 - AMs - Ear2, Itgax
2 - Endothelial (Probably microvascular/capillary endothelium) - Pecam1
3 - AT2 cells - Sftpd
4 - Neutrophils - Itgam, Ly6g, Pim1-hi
5 - Myofibroblasts (+ lipofibroblasts?) - Myh10, Col1a2, Inmt (lipofibroblast gene)
6 - Plasma cells - Cd19, Mzb1, Pim1
7 - CD8 T cells - Cd8a, Cd3e
8 - Matrix fibroblasts - Col14a1
9 - CD4 T cells - CD4, Cd3e
10 - Cx3cr1+ macrophages - Cx3cr1, Adgre4
11 - Car4+ endothelial
12 - F13a1+ monocytes - Ccr2, Ly6c2, Adgre1-
13 - **Mix of mulitple pure cell types - Neutrophils, B cells, AMs, ciliated cells - REMOVE**
14 - Mesothelial - Msln, Upk1b, Igfbp5
15 - cDCs (Mix of cDC1 and cDC2; see FeaturePlot of Xcr1 and Itgam) - Itgae, Xcr1, Ccl17
16 - Club cells - Scgb1a1, Bpifb1-lo, Sftpd-int
17 - Macrovascular endothelial - Pecam1, Vwf, Top2a, S.Score-hi
18 - Mitotic - Top2a
19 - Interstitial macrophages - C1qb, Cx3cr1, F13a1, Ly6c2-, Pf4
20 - Gamma/delta T cells - Cd3e, Trdc
21 - NK cells - Nkg7, Prf1
22 - Mitotic B cells - Cd19, Hells, S.Score-hi
23 - Ciliated cells - Foxj1, Rabl2, Cd24a, S.Score-lo
24 - **Impure/doublet mix - Neutrophils and endothelial cells - REMOVE** - S100a9, Pecam1
25 - Pericytes - Pdgfrb, Acta2, Myh11, Tagln
26 - Platelets - Pf4, Itga2, Ppbp (Cxcl7)
27 - Erythrocytes - Hba-a1
28 - Mitotic T cells (more CD8 T cells than CD4 T cells) - Cd3e, Trac, S.Score-hi
29 - Lymphatics - Prox1, Mmrn1
30 - Suppressive neutrophils - Ly6g, Cxcr2, Pim1, Cd274 (PD-L1), Fapb5
31 - Mucosal/IgA B cells - Igha, Cd19, Cd79a/b, Jchain
32 - Mast cells - Hdc, Fcer1a, Ccl3, Ccl4, Il6
33 - Goblet cells - Scgb1a1, Bpifb1, Cyp2f2, Sox2
34 - **Impure/doublet mix - T cells and endothelial cells - REMOVE** - Cd93, Itga1, Adgrf5, Pecam1, high nFeature_RNA
35 - Ccr7+ DCs - Ccl17, Ccl22, Ccr7, Cxcl16, Fabp5, Cxcl16, Marcks (Brown et al., 2019, Cell; doi: 10.1016/j.cell.2019.09.035)
36 - Fn1+ macrophages (name from Angelidis et al. - maybe just conventional macrophages?) - Itgam, Adgre1, Fn1, C1qb, Prg4, Wnt2, Pf4, Arg1
37 - AT1 cells - Hopx, Rtkn2, Vegfa


# Res 0.4 markers below, no longer using this resolution
#
Cluster - Identity - Genes (**Res 0.4**)
0 - Naive B cells - CD19, CD44-lo
1 - Endothelial (**Car4 neg, but which type? - probably Capillary**) - Pecam1
2 - AMs (**which type**) - Ear2, Itgax
3 - T cells - CD3e, CD4, CD8a
4 - AT2 cells - Sftpd (**contrast with C24**)
5 - Neutrophils - Itgam, Ly6g, Pim1-hi
6 - Myofibroblasts (+ lipofibroblasts?) - Myh10, Col1a2, Inmt (lipofibroblast gene)
7 - Plasma cells - Cd19, Mzb1
8 - Matrix fibroblasts - Col14a1
9 - 
10
11 - Car4+ Endothelial
12 - Club cells - Scgb1a1, Bpifb1-lo
13 - Goblet cells - Scgb1a1, Bpifb1
14 - Mesothelial - Msln, Upk1b
15
16 - Vwf+ endothelial
17
18
19 - Interstitial macrophages - C1qb, Cx3cr1, Fna/Prg4-neg
20
21
22 - Ciliated cells - Foxj1
23 - Pericytes - Pdgfrb, Notch3, Mcam, Cspg4, Pecam1-int
24 - AT2 cells (*+ RBCs?*) - Sftpd (**contrast with C4**), Hb1-a1
25 - Lymphatic - Mmrn1
26
27 - ? - Pim1-hi
28
29 - *Endothlial with some Ptprc expression?*
30 - Fn1+ macrophages (name from Angelidis et al. - maybe just conventional macrophages?) - Itgam, Fn1, C1qb, Prg4
#

```{r Markers - B cells}
### What's the difference among B cell clusters (0, 6, 22)? Esp. 0 and 22, very different freqs at D3 in Young vs Aged
Flu_markers_Bcell <- subset(Flu, idents = c(0, 4, 17 ,22)) %>% FindAllMarkers(only.pos = T)
Flu_markers_Bcell %>% group_by(cluster) %>% top_n(20, wt = avg_log2FC)
# Zbtb32 and Pim1 highly upregulated in cluster 6
# Flu_markers_C22 <- FindMarkers(Flu, ident.1 = 22, only.pos = T)
VlnPlot(Flu, idents = c(0, 6, 22), features = c("Ptprc", "Cd19", "Cd44", "Mzb1", "Zbtb32", "Pim1"), pt.size = 0)
VlnPlot(Flu, idents = c(0, 6, 22), features = c("Hells", "Dtl", "Clspn", "Uhrf1", "S.Score", "G2M.Score"), pt.size = 0)
# 0 is naive B cells (Cd44 lo), 7 is plasma cells (Mzb1), 21 is replicating (Hells, S.Score)

table(Flu$seurat_clusters, Flu$orig.ident)
# Note that the vast majority of plasma cells were recovered from D3_O2
```

```{r Markers - Endothelial}
DotPlot(Flu, features = c("Pecam1", "Vwf", "Car4", "Mmrn1", "Tmem100", "Ednrb", "Vcam1")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Car4-hi
DotPlot(Flu, features = c("Pecam1", "Car4", "Cd34")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Lymphatics
DotPlot(Flu, features = c("Prox1", "Mmrn1")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# C24
DotPlot(Flu, features = c("Pecam1", "Ace", "Adgrf5", "Cd93", "Itga1")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Macrovascular endothelium - C17
DotPlot(Flu, features = c("Pecam1", "Plvap", "Vwf", "Vcam1")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Microvascular endothelium - C2, C34
DotPlot(Flu, features = c("Pecam1", "Gpihbp1", "Ifi47", "Plvap", "Sox17")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# How is C34 different from C2?
# Flu_markers_C34_C2 <- FindMarkers(Flu, ident.1 = 34, ident.2 = 2)

# C34 - Doublets or contaminated mix of T cells and endothelial cells - REMOVE
DotPlot(Flu, features = c("Ptprc", "Cd3e", "Pecam1")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
FeaturePlot(subset(Flu, idents = 34), features = c("Ptprc", "Cd3e", "Pecam1"), ncol = 2)
VlnPlot(Flu, features = c("nFeature_RNA"), idents = c(2, 34), pt.size = 0)

# How is C24 different from C2?
# Flu_markers_C24_C2 <- FindMarkers(Flu, ident.1 = 24, ident.2 = 2)

# C34 - Doublets or contaminated mix of neutrophils and endothelial cells - REMOVE
DotPlot(Flu, features = c("Cdh5", "Cldn5", "Itga1", "Cd93", "Ace")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
FeaturePlot(subset(Flu), features = c("Cxcl2", "Pecam1", "Ptprc", "Itgam"), ncol = 2)
VlnPlot(Flu, features = c("nFeature_RNA"), idents = c(2, 24), pt.size = 0)
```

```{r Markers - Epithelial/AT2/AT1/goblet/club cells}
DotPlot(Flu, features = c("Ptprc", "Scgb1a1", "Sftpd", "Rtkn2", "Sox2", "Foxj1", "Bpifb1", "Cftr", "Rabl2", "Cyp2f2", "Cd24a")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Ciliated cells
DotPlot(Flu, features = c("Sec14l3", "Rsph1", "Foxj1")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# What's the difference between C13 and C23? Both seem to be ciliated-like.
# Flu_markers_C13_C23 <- FindMarkers(Flu, ident.1 = 13, ident.2 = 23)
DotPlot(Flu, features = c("Sec14l3", "Rsph1", "Foxj1", "Sftpa1", "Sftpc", "Ear2")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
FeaturePlot(subset(Flu, idents = 25), features = c("Ptprc", "Sec14l3"), ncol = 2)
DimPlot(subset(Flu, idents = 13))
# C13 looks like a mix of B cells, ciliated cells, AMs, and neutrophils all stuck in the same cluster
# May be simpler to exclude this cluster as contamination instead of increasing resolution any further

# Mesothelial
DotPlot(Flu, features = c("Msln", "Upk1b")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# AT2
DotPlot(Flu, features = c("Sftpd", "Sftpb", "Sftpc", "Cxcl15", "Lamp3")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Club and goblet cells
DotPlot(Flu, features = c("Scgb1a1", "Sox2", "Cyp2f2", "Bpifb1")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# AT1
DotPlot(Flu, features = c("Hopx", "Rtkn2", "Vegfa")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r Markers - Fibroblasts}
DotPlot(Flu, features = c("Pdgfrb", "Acta2", "Myh10", "Myh11", "Tagln", "Cldn5", "Pdgfra", "Cdh5", "Col1a1", "Col1a2", "Col4a1", "Col13a1", "Col14a1", "Mmp3", "Pi16", "Cygb", "Rtp4", "Pecam1", "Wnt2", "Fgf10", "Msln")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
# Pdgfrb is a pericyte marker (Villa Ellis et al.)
# Acta2a, Myh11, Tagln mark myofibroblasts (10.1016/j.celrep.2018.03.010)

DotPlot(Flu, features = c("Fabp1", "Fabp4", "Fabp5", "Lpl", "Lipa", "Inmt", "Wnt2")) # Lipofibroblast
DotPlot(Flu, features = c("Col14a1", "Mmp3", "Pi16", "Cygb", "Rtp4")) # Col14a1-hi matrix fibroblast markers (Xie et al., 10.1016/j.celrep.2018.03.010)
DotPlot(Flu, features = c("Pdgfrb", "Notch3", "Mcam", "Cspg4")) # Pericyte markers (Xie et al., 10.1016/j.celrep.2018.03.010)

FeaturePlot(subset(Flu, idents = 5), features = c("Myh10", "Inmt"))
```

```{r Markers - Myeloid}
# General myeloid
DotPlot(Flu, features = c("Ptprc", "Itgam", "Itgax", "Itgae", "Ear2", "Adgre1", "Cx3cr1", "C1qb", "Prg4", "Fn1", "Ly6c2", "Ly6g", "Lyz2")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Neutrophils
DotPlot(Flu, features = c("Ptprc", "Itgam", "Cxcr2", "Ly6g", "Csf3r", "Pim1", "Cd274")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
# Neutrophil and suppressive neutrophil genes split
DotPlot(Flu, features = c("Ptprc", "Itgam", "Cxcr2", "Ly6g", "Pim1", "Arg1", "Cd274"), idents = 4, group.by = "Timepoint_Age")
# Pim1 and PD-L1 (Cd274) most highly expressed on D3

# Supressive neutrophils
DotPlot(Flu, features = c("Ptprc", "Itgam", "Cxcr2", "Ly6g", "Pim1", "Cd274", "Stat3", "Arg1", "Arg2")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
VlnPlot(Flu, features = c("Fabp5", "Cxcr2", "Csf3r", "S100a9", "Pim1", "Cd274"), idents = c(8, 27), pt.size = 0)
VlnPlot(Flu, features = c("Fabp5", "Fgl2", "Clec4d", "Il1b", "Pim1", "Cd274"), idents = c(8, 27), pt.size = 0)

# DCs
DotPlot(Flu, features = c("Ptprc", "Itgax", "Itgae", "Itgam", "Ccl17", "Xcr1", "Ccl22", "Tbx21", "Rorc", "Ccr7")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
FeaturePlot(subset(Flu, idents = 15), features = c("Itgax", "Itgam", "Xcr1"), ncol = 2)

# Monocytes
DotPlot(Flu, features = c("Ptprc", "Itgam", "Ly6c2", "Cx3cr1", "F13a1", "Ccr2", "Lyz2", "Adgre1", "Adgre4")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Mast cells/basophils/esoinophils
DotPlot(Flu, features = c("Ptprc", "Itgam", "Hdc", "Fcer1a", "Tpsab1")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# AMs
DotPlot(Flu, features = c("Ptprc", "Itgam", "Itgax", "Adgre1", "Adgre4", "Ear2")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r Markers - Lymphocytes}
DotPlot(Flu, features = c("Ptprc", "Cd19", "Cd79a", "Cd79b", "Jchain", "Igha", "Cd3e", "CD4", "Cd8a", "Trdc", "Nkg7")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(Flu, features = c("CD4", "Cd8a", "S.Score", "G2M.Score"), idents = c(7, 9, 20, 21, 28), ncol = 2, pt.size = 0)
DotPlot(Flu, features = c("CD4", "Cd8a", "Trdc", "Fcgr3", "S.Score", "G2M.Score"), idents = c(7, 9, 20, 21, 28))
```

```{r Markers - Other}
# RBCs
DotPlot(Flu, features = c("Hba-a1", "Hbb-bs")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Platelets
DotPlot(Flu, features = c("Pf4", "Itga2b", "Ppbp")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Neuroendocrine
DotPlot(Flu, features = c("Cpa3", "Calca", "Slc18a2")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
FeaturePlot(subset(Flu, idents = 32), features = c("Cpa3", "Slc18a2", "Ptprc", "Hdc"))
# C32 is a mix of mast cells and neuroendocrine, but they do separate cleanly; higher resolution could tease these apart, but not necessary for this study

# Mitotic
DotPlot(Flu, features = c("S.Score", "G2M.Score", "Top2a")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
VlnPlot(Flu, features = c("S.Score", "G2M.Score", "Top2a"), pt.size = 0, ncol = 2)
# C18 (mix of cells or endothelial specifically?), C22 (B cells), C28 (T cells), and C34 are all mitotic
```

```{r Subset}
Flu_pure <- subset(Flu, idents = c(30), invert = T)

dim(Flu_pure) # 35414 47209
table(Flu_pure$seurat_clusters)
```
Cluster - Identity - Genes (**Res 1**) - RJB
0 - Naive B cells
1 - Endothelial cells
2 - CD8 T cells
3 - Aveolar Macrophages
4 - Intersitial macrophages
5 - Monocytes
6 - CD4 T cells
7 - Mitotic B cells
8 - Neutrophils
9 - Macrophages
10 - Club cells
11 - Plasma cells
12 - Matrix Fibtroblasts
13 - Car4+ endothelial cells
14 - NK
15 - Mitotic B cells
16 - Platelets
17 - Myofibroblasts
18 - cDC
19 - Vwf+ endothelial cells
20 - Mitotic T cells
21 - Goblet cells
22 - Mesothelial cells
23 - AT2
24 - Gamma delta T cells
25 - Myofibroblasts
26 - Cilliated cells
27 - Supressive Neutrophils
28 - Lymphatic endothelial cells
29 - RBC
30 - **JUNK**
31 - Myofibroblasts
32 - Ccr7+ DC
33 - Mitotic B cells
34 - Mitotic
35 - Pericytes
36 - Mast Cells
37 - Ccr9 - remove
38 - IghA+ B cells
39 - 




```{r Rename clusters}
Flu <- RenameIdents(Flu, 
                    "0" = "Endothelial cells",
                    "1" = "Naive B cells",
                    "2" = "CD8 T cells",
                    "3" = "Intersitial macrophages",
                    "4" = "Alveolar macrophages",
                    "5" = "Monocytes",
                    "6" = "Mitotic B cells",
                    "7" = "Neutrophils",
                    "8" = "Macrophages",
                    "9" = "Club cells",
                    "10" = "Plasma cells",
                    "11" = "Matrix Fibtroblasts",
                    "12" = "CD4 T cells",
                    "13" = "Car4+ endothelial cells", 
                    "14" = "NK",
                    "15" = "Myofibroblasts",
                    "16" = "Mitotic B cells",
                    "17" = "Platelets",
                    "18" = "Alveolar macrophages",
                    "19" = "cDC",
                    "20" = "CD4 T cells",
                    "21" = "Vwf+ endothelial cells",
                    "22" = "Mitotic T cells",
                    "23" = "Goblet cells",
                    "24" = "Mesothelial cells", 
                    "25" = "Gamma delta T cells",
                    "26" = "Myofibroblasts",
                    "27" = "Cilliated cells",
                    "28" = "AT2",
                    "29" = "Plasma cells",
                    "30" = "Lymphatic endothelial cells",
                    "31" = "RBC",
                    "32" = "Myofibroblasts",
                    "33" = "33",
                    "34" = "Mitotic B cells", 
                    "35" = "Ccr7+ DC",
                    "36" = "Suppressive neutrophils", 
                    "37" = "Pericytes",
                    "38" = "Mast cells",
                    "39" = "39",
                    "40" = "IgA+ B cells",
                    "41" = "Alveolar macrophages",
                    "42" = "42",
                    "43" = "Alveolar macrophages",
                    "44" = "Neutrophils",
                    "45" = "AT1")

DimPlot(Flu_pure, label = T) +
  NoLegend()

Flu_pure <- RenameIdents(Flu_pure, 
                    "Supressive Neutrophils" = "Suppressive neutrophils")
Flu_pure <- RenameIdents(Flu_pure, 
                    "Aveolar macrophages" = "Alveolar macrophages")
Flu_pure <- RenameIdents(Flu_pure, 
                    "Intersitial macrophages" = "Interstitial macrophages")
Flu_pure <- RenameIdents(Flu_pure, 
                    "Matrix Fibtroblasts" = "Matrix Fibroblasts")

Flu_pure <- RenameIdents(Flu_pure, 
                    "Matrix Fibroblasts" = "Matrix fibroblasts")
Flu_pure <- RenameIdents(Flu_pure, 
                    "CD8 T cells" = "CD8 T cells")
Flu_pure <- RenameIdents(Flu_pure, 
                    "CD4 T cells" = "CD4 T cells")
table(Flu_pure@active.ident)

names(sort(table(Idents(Flu_pure)), decreasing = T))


Flu_pure <- RenameIdents(Flu_pure, 
                         "Alveolar macrophages" = "Alveolar macrophages",
                         "Mitotic B cells" = "Mitotic B cells",
                    "Endothelial cells" = "Endothelial cells",
                    "Naive B cells" = "Naive B cells",
                    "CD8 T cells" = "CD8 T cells",
                    "Interstitial macrophages" = "Interstitial macrophages",
                    "CD4 T cells" = "CD4 T cells",
                    "Monocytes" = "Monocytes",
                    "Neutrophils" = "Neutrophils",
                    "Plasma cells" = "Plasma cells",
                    "Myofibroblasts" = "Myofibroblasts",
                    "Macrophages" = "Macrophages",
                    "Club cells" = "Club cells",
                    "Matrix fibroblasts" = "Matrix fibroblasts",
                    "Car4+ endothelial cells" = "Car4+ endothelial cells", 
                    "NK" = "NK",
                    "Platelets" = "Platelets",
                    "cDC" = "cDC",
                    "Vwf+ endothelial cells" = "Vwf+ endothelial cells",
                    "Mitotic T cells" = "Mitotic T cells",
                    "Goblet cells" = "Goblet cells",
                    "Mesothelial cells" = "Mesothelial cells", 
                    "Gamma delta T cells" = "Gamma delta T cells",
                    "Cilliated cells" = "Cilliated cells",
                    "AT2" = "AT2",
                    "Lymphatic endothelial cells" = "Lymphatic endothelial cells",
                    "RBC" = "RBC",
                    "Ccr7+ DC" = "Ccr7+ DC",
                    "Suppressive neutrophils" = "Suppressive neutrophils", 
                    "Pericytes" = "Pericytes",
                    "Mast cells" = "Mast cells",
                    "IgA+ B cells" = "IgA+ B cells",
                    "AT1" = "AT1")

saveRDS(Flu_pure, "RDS_files.nosync/Flu_pure_RJB.RDS")
Flu_pure <-readRDS("RDS_files.nosync/Flu_pure_RJB.RDS")

```

```{r}
Flu_pure <- subset(Flu, idents = c(39,33,42), invert = T) # No Fn1+ macrophage cluster
```



```{r Save Load}
saveRDS(Flu_pure, "RDS_files.nosync/Flu_D0_D3_D9_pure_RJB.RDS")
Flu_pure <- readRDS("RDS_files.nosync/Flu_D0_D3_D9_pure_RJB.RDS")
dim(Flu_pure)
Flu_pure$Named_clusters <- Flu_pure@active.ident
Flu_pure$Cluster_Time_Age <- paste0(Flu_pure$Named_clusters, " ", Flu_pure$Timepoint_Age)
Flu_pure$Cluster_Time_Age <- factor(Flu_pure$Cluster_Time_Age,
                                    levels = intersect(sapply(levels(Flu_pure$Named_clusters), FUN = function(x) {paste0(x, " ", c("Day 0 Young", "Day 0 Aged", "Day 3 Young", "Day 3 Aged", "Day 9 Young", "Day 9 Aged"))}) %>% as.character(), unique(Flu_pure$Cluster_Time_Age)))
```

```{r New UMAPs}
DimPlot(Flu_pure, label = T, repel = T, label.size = 4.5) +
  NoLegend()
ggsave("Figures_Flu_pure_RJB/01_UMAP.png", width = 12, height = 7)

DimPlot(Flu_pure, split.by = "orig.ident", ncol = 5)
ggsave("Figures_Flu_pure_RJB/01_UMAP_split_origident.png", width = 15, height = 7)

DimPlot(Flu_pure, split.by = "Timepoint_Age", ncol = 2)
ggsave("Figures_Flu_pure_RJB/01_UMAP_split_TimepointAge.png", width = 12, height = 7)

DimPlot(Flu_pure, group.by = "orig.ident", order = F, pt.size = 0.001)
ggsave("Figures_Flu_pure_RJB/01_UMAP_groupby_origident.png", width = 12, height = 7)

DimPlot(Flu_pure, group.by = "Phase")
```

```{r Save legend for SPOTlight}
# Save scRNA-seq legend for SPOTlight, these use the same cell types **but** SPOTlight is in alphabetical order

Flu_pure$SPOTlight_levels <- Flu_pure$Named_clusters %>% as.character()

as_ggplot(
  get_legend(
    DimPlot(subset(Flu_pure, idents = c("RBC"), invert = T), group.by = "SPOTlight_levels") +
      guides(color = guide_legend(ncol = 3, override.aes = list(size = 2)))
    ))
ggsave("Figures_Flu_pure_RJB/01_SPOTlight_legend.pdf", width = 6.75, height = 2.75)

# Flu_pure$SPOTlight_levels <- NULL
```

```{r Cluster breakdown}
table(Flu_pure@active.ident, Flu_pure$orig.ident) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency")
ggsave("Figures_Flu_pure_RJB/02_Cluster_barplot_origident.pdf", width = 10, height = 5)

table(Flu_pure@active.ident, Flu_pure$Timepoint) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  xlab("Timepoint") +
    geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 0.025,
        paste0(Cluster),
        "")),
    position = position_stack(vjust = 0.5),
    size = 3)
ggsave("Figures_Flu_pure_RJB/02_Cluster_barplot_Timepoint.pdf", width = 10, height = 5)

table(Flu_pure@active.ident, Flu_pure$Age) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  xlab("Age") +
    geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 0.025,
        paste0(Cluster),
        "")),
    position = position_stack(vjust = 0.5),
    size = 4)
ggsave("Figures_Flu_pure_RJB/02_Cluster_barplot_Age.pdf", width = 10, height = 5)

table(Flu_pure@active.ident, Flu_pure$Age) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank(), legend.position = "none") +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  xlab("Age") +
    geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 0.025,
        paste0(Cluster),
        "")),
    position = position_stack(vjust = 0.5),
    size = 3.5)
ggsave("Figures_Flu_pure_RJB/02_Cluster_barplot_Age_NoLegend.pdf", width = 5, height = 5)

table(Flu_pure@active.ident, Flu_pure$Timepoint_Age) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
    geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 0.025,
        paste0(Cluster),
        "")),
    position = position_stack(vjust = 0.5),
    size = 1.9)
ggsave("Figures_Flu_pure_RJB/02_Cluster_barplot_TimepointAge.pdf", width = 10, height = 5)


write.xlsx2(table(Flu_pure@active.ident, Flu_pure$Age) %>%
  prop.table(margin = 2) * 100, "Figures_Flu_pure_RJB/02_table1.xlsx", row.names = TRUE)                                   

table(Flu_pure@active.ident, Flu_pure$Timepoint_Age) %>%
  prop.table(margin = 2) * 100

table(Flu_pure@active.ident, Flu_pure$orig.ident) %>%
  prop.table(margin = 2) * 100
```

```{r Final cluster marker dotplot}
DefaultAssay(Flu_pure) <- "RNA"

DotPlot(Flu_pure, features = c("Cd19", "Ear2", "Pecam1", "Sftpd", "Ly6g", "Myh10", "Mzb1", "Cd8a", "Col14a1", "CD4", "Cx3cr1", "Car4", "Ly6c2", "Msln", "Itgae", "Scgb1a1", "Vwf", "Top2a", "C1qb", "Trdc", "Nkg7", "Hells", "Foxj1", "Pdgfrb", "Ppbp", "Hba-a1", "Prox1", "Cd274", "Igha", "Fcer1a", "Bpifb1", "Ccr7", "Fn1", "Hopx")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title = element_blank())


DotPlot(Flu_pure, features = c( "Cd4","Cd8a","Col14a1","C1qb","Ear2","Cd274","Pecam1", "Cd19", "Ly6c2","Hells","Ly6g","Cx3cr1", "Scgb1a1", "Mzb1",    "Car4",  "Nkg7",  "Myh10",  "Ppbp","Itgae","Vwf","Top2a", "Bpifb1", "Msln", "Trdc", "Foxj1", "Sftpd","Prox1", "Hba-a1", "Ccr7", "Pdgfrb", "Fcer1a","Igha", "Hopx")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title = element_blank())
ggsave("Figures_Flu_pure_RJB/03_Cluster_marker_DotPlot.pdf", width = 11, height = 7)


### Other random plots
# Weiguo - Cd11c+ Tbet+ B cells? (new subset found by Joe Craft - not here, probably only in lymph nodes)
DotPlot(Flu_pure, features = c("Cd19", "Igha", "Itgax", "Tbx21"))
```

```{r Random markers}
# Epithelial cells express Epcam
DotPlot(Flu_pure, features = c("Epcam"))
```

```{r Senecscence genes}
DefaultAssay(Flu_pure) <- "RNA"

Geneset_senescence <- readLines("Gene_sets/REACTOME_CELLULAR_SENESCENCE.txt")[c(-1, -2)] # Remove first two lines, which are headers
Geneset_senescence <- paste0(
 substring(Geneset_senescence, 1, 1),
 tolower(substring(Geneset_senescence, 2))) # Change capitalization
Geneset_senescence <- Geneset_senescence[which(Geneset_senescence %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_senescence <- list(Geneset_senescence)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_senescence, name = "Geneset_senescence")


VlnPlot(Flu_pure, features = c("Cdkn1a"), group.by = "Age", pt.size = 0, cols = c("forestgreen", "purple3")) +
  stat_compare_means(
    comparisons = list(c("Young", "Aged")),
    label.y = 5.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  ylim(-0.05, 6)
ggsave("Figures_Flu_pure_RJB/04_Cdkn1a_VlnPlot.pdf", width = 5, height = 3.5)

VlnPlot(Flu_pure, features = c("Geneset_senescence1"), group.by = "Age", pt.size = 0, cols = c("forestgreen", "purple3")) +
  stat_compare_means(
    comparisons = list(c("Young", "Aged")),
    label.y = 0.375) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title.x = element_blank(),
        legend.position = "none") +
  ylim(-0.15, 0.425) +
  ggtitle("Reactome Cellular Senescence gene set")
ggsave("Figures_Flu_pure_RJB/04_Module_Senescence_Age_VlnPlot.pdf", width = 5, height = 3.5)
```

```{r Cdkn1a}
DotPlot(Flu_pure, features = c("Cdkn1a")) + coord_flip() + RotatedAxis()
ggsave("Figures_Flu_pure_RJB/04_Cdkn1a_DotPlot.pdf", width = 10, height = 5)


VlnPlot(Flu_pure, features = c("Cdkn1a"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5, 7.5, 7)) +
  ylim(-0.01, 8) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/04_Time_Age_Cdkn1a_VlnPlot.pdf", width = 5, height = 3.5)


VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")), features = c("Cdkn1a"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(6,6, 6)) +
  ylim(-0.01, 8) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") + ggtitle("Car4+ endothelial Cells Cdkn1a Expression")
ggsave("Figures_Flu_pure_RJB/04_04_Car4_Cdkn1a_VlnPlot.pdf")



VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")), features = c("Cdkn1a"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(6,6, 6)) +
  ylim(-0.01, 8) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") + ggtitle("Endothelial Cells Cdkn1a Expression")
ggsave("Figures_Flu_pure_RJB/04_04_Car4_Cdkn1a_VlnPlot.pdf")




```


```{r DEGs by Age}
DefaultAssay(Flu_pure) <- "RNA"

# Total DEGs in Young and Old

# Flu_pure@active.ident <- Flu_pure$Age
# Flu_pure_DEGs_Age <- FindAllMarkers(Flu_pure, only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters

# write.csv(Flu_pure_DEGs_Age, "Flu_pure_DEGs_Age.csv")
Flu_pure_DEGs_Age <- read.csv("Flu_pure_DEGs_Age.csv", row.names = 1)
Flu_pure_DEGs_Age %>% group_by(cluster) %>% top_n(30, wt = avg_log2FC)

# Not that helpful, really just identifies which sample has higher numbers of certain cell types
```

```{r Neutrophils - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_Neutro_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Neutrophils")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
#Flu_pure_DEGs_Neutro_TimeAge %>% group_by(cluster) %>% top_n(20, wt = avg_log2FC)

DotPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Ltf", "Ngp", "Camp", "Il1b", "Il1a", "Ccl3", "Cxcl3", "Cxcl1", "Cd274", "Bcl2a1b", "Traf1", "Sod2", "Acod1", "Fosl2", "Fosb", "Nr4a1", "Myadm","Cebpd", "Gpx4", "Plk3", "Cd300lb"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title = element_blank()) +
  coord_flip()
ggsave("Figures_Flu_pure_RJB/05_Neutrophils_DEGs_DotPlot.pdf")

# Ltf (lactotransferrin) - 0 - neutrophil granule protein, iron-binding, found in mucosal secretions
# Camp (cathelicidin antimicrobial peptide) - D0 Young - stored in neutrophil granules, target of vitamin D3 receptor (10.1096/fj.04-3284com)
# Ngp (neutrophilic granule protein)
# Shfm1 (26S proteasome subunit) - D0 Old - neutrophil function unknown
# Ccl3 (MIP-1a; Ccr1/4/5 ligand) - D3 Young - Recruits DCs (10.1371/journal.ppat.1000755) and neutrophils - locus opens once neutrophils reach tissue from blood (10.1038/s41590-021-00968-4)
# Bcl2a1b - D3 Y=O - Anti-apoptotic gene (10.1038/cdd.2017.30)
# Cd274 (PD-L1) - D3 Y > O
# Traf1 - D3 Young = Old - Anti-inflammatory, downstream of TNF signaling, inhibits Nemo (IKKg) ubiquitination
# Sod2 (Superoxide dismutase 2) - D3 Young = Old
# Acod1 (Aconitate decarboxylase 1 aka IRG1) - D3 Young, D9 Old - upregulated downstream of TLRs, IFNAR, MYD88, NF-kB, STATs, IRFs (10.1038/s41423-020-0489-5)
# Nr4a1 - XYZ function in neutrophis? Not much found on quick lit search.
# Myadm (Myeloid-associated differentiation marker) - D9 Young, D9 Old - Locus accessibility increases when neutrophils are released into blood from bone marrow (10.1038/s41590-021-00968-4)
# Hspa8/1a/1b - D9 Young
# Cebpd - XYZ
# Gpx4 (glutathione peroxidase 4) - D3 Young, D9 Old - maybe upregulated in response to oxidative stress? Check ferroptosis GSEA (10.1038/s41590-021-00993-3)
# Plk3 (Polo-like kinase 3) - D9 Old - induces apoptosis in p53-dependent and -independent manner
# Cd300lb - D9 Old = D9 Young - Binds phosphatidylserine on apoptotic cells to trigger phagocytosis; function enhanced when bound to DAP12 (Tyrobp) (10.4161/23723548.2014.964625 and 10.1038/cdd.2014.86)

DotPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Cxcr2", "Ccr1"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Traf1"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(c("Day 3 Young", "Day 3 Aged"))) +
  ylim(-0.5, 6)

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Sod2"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(c("Day 3 Young", "Day 3 Aged"))) +
  ylim(-0.5, 6)

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Acod1"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 7) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Plk3"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 7) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Cxcl2"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5, 7.5, 7)) +
  ylim(-0.01, 8) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Cxcl2 - Neutrophils")
ggsave("Figures_Flu_pure_RJB/05_Neutrophils_Cxcl2_VlnPlot.pdf")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Cxcr2"), group.by = "Timepoint_Age", pt.size =0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5, 7.5, 7)) +
  ylim(-0.01, 8) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Cxcr2 - Neutrophils")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Cd274"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5, 7.5, 7)) +
  ylim(-0.01, 8) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Ccl6"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5, 7.5, 7)) +
  ylim(-0.01, 8) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Ccl6 - Neutrophils")



# Neutrophil and suppressive neutrophil PD-L1 expression (Arg1 too low to be useful)
VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Cd274"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.5, 4.5, 4.5)) +
  ylim(-0.01, 5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Cd274 (PD-L1) - Neutrophils")

VlnPlot(subset(Flu_pure, idents = c("Suppressive neutrophils")), features = c("Cd274"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(3.5, 3.5)) +
  ylim(-0.01, 4) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Cd274 (PD-L1) - Suppressive neutrophils")

# ROS production module score
Geneset_ROS <- readLines("Gene_sets/HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY.txt")[c(-1, -2)] # Remove first two lines, which are headers
Geneset_ROS <- paste0(
 substring(Geneset_ROS, 1, 1),
 tolower(substring(Geneset_ROS, 2))) # Change capitalization
Geneset_ROS <- Geneset_ROS[which(Geneset_ROS %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_ROS <- list(Geneset_ROS)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_ROS, name = "Geneset_ROS")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Geneset_ROS1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.5, 0.5, 0.5)) +
  ylim(-0.2, 0.6) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Hallmark Reactive Oxygen Species gene set - Neutrophils")
ggsave("Figures_Flu_pure_RJB/05_Neutrophils_ROS_Module_VlnPlot.pdf")

# Inflammation module score
Geneset_Inflammation <- readLines("Gene_sets/HALLMARK_INFLAMMATORY_RESPONSE.txt")[c(-1, -2)] # Remove first two lines, which are headers
Geneset_Inflammation <- paste0(
 substring(Geneset_Inflammation, 1, 1),
 tolower(substring(Geneset_Inflammation, 2))) # Change capitalization
Geneset_Inflammation <- Geneset_Inflammation[which(Geneset_Inflammation %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_Inflammation <- list(Geneset_Inflammation)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Inflammation, name = "Geneset_Inflammation")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Geneset_Inflammation1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.25, 0.4, 0.4)) +
  ylim(-0.1, 0.45) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Hallmark Inflammatory Response gene set - Neutrophils")
ggsave("Figures_Flu_pure_RJB/05_Neutrophils_Inflammatory_Module_VlnPlot.pdf")

DotPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Il1b", "Tnf", "Il1a"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Apoptosis module score
Geneset_Apoptosis <- readLines("Gene_sets/HALLMARK_APOPTOSIS.txt")[c(-1, -2)] # Remove first two lines, which are headers
Geneset_Apoptosis <- paste0(
 substring(Geneset_Apoptosis, 1, 1),
 tolower(substring(Geneset_Apoptosis, 2))) # Change capitalization
Geneset_Apoptosis <- Geneset_Apoptosis[which(Geneset_Apoptosis %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_Apoptosis <- list(Geneset_Apoptosis)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Apoptosis, name = "Geneset_Apoptosis")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Geneset_Apoptosis1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.275, 0.4, 0.4)) +
  ylim(-0.1, 0.45) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Hallmark Apoptosis gene set - Neutrophils")
ggsave("Figures_Flu_pure_RJB/05_Neutrophils_Apoptosis_Module_VlnPlot.pdf")

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Il1b"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5, 5, 5)) +
  ylim(-0.05, 5.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Neutrophils - NETosis}
# Jie Sun asked if there's a difference in NETosis in netrophils between young and aged mice

# Couldn't find a premade GSEA NETosis gene list, so using one from a human diabetes paper (10.2337/db15-0863)

Flu_pure <- AddModuleScore(Flu_pure,
                           features = list(c("Prtn3", "Hist1h4a", "Elane", "Lcn2", "Mpo", "Ltf", "Hist1h2h", "Actn4", "Hist1h2bk", "Lcp1", "Flna", "H3f3c", "Tkt", "Pfn1", "Myh9", "Cat", "Gpi", "Gapdh", "Eno1", "Anxa1", "Msn", "Actr", "Krt10", "S100a8", "S100a9")),
                           name = "Geneset_NETosis")
# The following features are not present in the object: Hist1h2h, H3f3c, Gpi, Actr, not searching for symbol synonyms

VlnPlot(subset(Flu_pure, idents = c("Neutrophils")), features = c("Geneset_NETosis1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(1.9, 1.9, 1.9)) +
  ylim(-0.15, 2) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("NETosis gene set - Neutrophils")
```

```{r Neutrophil chemokine production}
# XYZ - Which cells make Cxcl2, Cxcl12 and Ccl5 to draw in neutrophils?
DotPlot(Flu_pure, features = c("Cxcl1", "Cxcl2", "Cxcl5", "Cxcl12", "Ppbp", "Ccl5"))
```

```{r Suppressive Neutrophils - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_GMDSC_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Suppressive neutrophils")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_GMDSC_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)

DotPlot(subset(Flu_pure, idents = c("Suppressive neutrophils")), features = c("Ccl3", "Cxcl3", "Il10", "Tgfb1", "Cd274", "Arg1", "Arg2", "Il1a", "Il1b", "Olr1", "Mpo"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Ccl3 - D3 Young
# Cxcl3 - D3 Y=O but D9 O > Y
# Tgfb1 - D3 Young
# Cd274 (PD-L1) - D3 Y=O; D9 Y=O; D3 > D9
# Arg1 - D0 Old
# Il1b - D3 O > Y


VlnPlot(subset(Flu_pure, idents = c("Suppressive neutrophils")), features = c("Tgfb1"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Suppressive neutrophils")), features = c("Tgfbr2"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 2.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Alveolar macrophages - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_AM_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Alveolar macrophages")), only.pos = T)
# Flu_pure_DEGs_AM_D0 <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("Alveolar macrophages") & Timepoint == "Day 0"), only.pos = T)
# Flu_pure_DEGs_AM_D3 <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("Alveolar macrophages") & Timepoint == "Day 3"), only.pos = T)
# Flu_pure_DEGs_AM_D9 <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("Alveolar macrophages") & Timepoint == "Day 9"), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_AM_TimeAge %>% group_by(cluster) %>% top_n(30, wt = avg_log2FC)
Flu_pure_DEGs_AM_D0 %>% group_by(cluster) %>% top_n(30, wt = avg_log2FC)
Flu_pure_DEGs_AM_D3 %>% group_by(cluster) %>% top_n(30, wt = avg_log2FC)
Flu_pure_DEGs_AM_D9 %>% group_by(cluster) %>% top_n(30, wt = avg_log2FC)

# Fewer AMs in Aged due to lower proliferation; changes in the lung ECM cause AMs to lose responsesiveness to GM-CSF (10.1172/JCI140299)

DotPlot(subset(Flu_pure, idents = c("Alveolar macrophages")), features = c("Cbr2", "Cd63", "Fn1", "Cxcl1", "Tnf", "Il1a", "Ccl4", "Nfkbia", "Cxcl10", "Nfkbiz", "Ereg", "Ccl5", "C3", "Cxcl2", "Gpx4", "Cybb", "Cd74", "H2-Aa", "H2-Ab1", "H2-Eb1"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title = element_blank()) +
  coord_flip()
ggsave("Figures_Flu_pure_RJB/07_AM_DEGs_DotPlot.pdf")

# Fn1 - Secreted by MDSC-like macrophages in PDAC to promote immunosuppression (10.1016/j.canlet.2020.11.013)
# Cxcl2 - D3 Y=O, D9 Old > Y
# Cd74 - D0/3/9 Old; Young increases from D3 to D9, Old stays constantly high (but increases from D0)
# Ereg - D3 Young ~= Aged; EGFR ligand, may promote contraleteral lung tissue growth after pneumonectomy (10.1002/jcp.24009), suggesting a wound-healing like interaction between AMs and stromal cells; equal expression in young and aged (after adjust p value) but Ereg almost exclusively expressed by AMs, predicted to signal more in Young by CellChat, and is higher in D3 Young by bulk RNA-seq
# Cybb (aka NOX2 aka NADPH oxidase 2, generates superoxide) - D3/9 Old - Acts in a cell-intrinsic immunosuppressive manner in AMs, with NOX2 ablation promoting inflammation (10.1182/blood.2021015365)
# Cd80 - D3 Young
# Nfkbia/z - Same at D3 but O > Y on D9
# Cd63 - Tetraspannin that acts as an exosome/extracellular vesicle marker (secreted to send proteins, lipids, lnc/mi/mRNA, DNA, small molecule metabolites to other cells) (10.3389/fmed.2020.618506; 10.3389/fimmu.2018.00924)

DotPlot(Flu_pure, features = c("Ereg", "Areg")) +
  theme(axis.title = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  coord_flip()
ggsave("Figures_Flu_pure_RJB/07_AM_Ereg_DotPlot.pdf", width = 18, height = 5)
# Ereg almost exclusively expressed by AMs

# What about Egfr (Ereg receptor)?
DotPlot(subset(Flu_pure, idents = c("Matrix fibroblasts", "Mesothelial cells", "AT1 cells")),
        features = c("Egfr"), group.by = c("Cluster_Time_Age"))



DotPlot(subset(Flu_pure, idents = c("Alveolar macrophages")), features = c("Cd80", "Cd86"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(subset(Flu_pure, idents = c("Alveolar macrophages")), features = c("Ereg"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Alveolar macrophages")), features = c("Cd63"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Alveolar macrophages")), features = c("S.Score"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.21, 0.21, 0.21)) +
  ylim(-0.2, 0.24) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Alveolar macrophages")), features = c("G2M.Score"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.21, 0.21, 0.21)) +
  ylim(-0.2, 0.24) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r AM-epithelial EGFR signaling}
# CellChat predicts decreased Ereg/Areg-Egfr signaling from AMs to epithelial cells (matrix fibroblasts, mesothelial, AT1) in aged vs young
# Use module scores of EGFR to check

Geneset_EGFR <- read_tsv("Gene_sets/BIOCARTA_EGF_PATHWAY.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_EGFR <- paste0(
 substring(Geneset_EGFR, 1, 1),
 tolower(substring(Geneset_EGFR, 2))) # Change capitalization
Geneset_EGFR <- Geneset_EGFR[which(Geneset_EGFR %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_EGFR <- list(Geneset_EGFR)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_EGFR, name = "Geneset_EGFR")

DotPlot(subset(Flu_pure, idents = c("Matrix fibroblasts", "Mesothelial cells", "AT1 cells")),
        features = c("Geneset_EGFR1"), group.by = "Cluster_Time_Age")

VlnPlot(subset(Flu_pure, idents = c("Matrix fibroblasts")), features = c("Geneset_EGFR1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.55, 0.55, 0.55)) +
  ylim(-0.35, 0.6) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Mesothelial cells")), features = c("Geneset_EGFR1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.55, 0.55, 0.55)) +
  ylim(-0.35, 0.6) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("AT1 cells")), features = c("Geneset_EGFR1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.55, 0.55, 0.55)) +
  ylim(-0.35, 0.6) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r AMs - Cell cycle}
# Fewer AMs in Aged due to lower proliferation (10.1172/JCI140299)

(table(subset(Flu_pure, idents = "Alveolar macrophages")$Phase,
      subset(Flu_pure, idents = "Alveolar macrophages")$Timepoint_Age) %>% 
  as.matrix() %>% 
  chisq.test(x = .))$p.value
# p = 5.57471e-41

table(subset(Flu_pure, idents = "Alveolar macrophages")$Phase,
      subset(Flu_pure, idents = "Alveolar macrophages")$Timepoint_Age) %>% 
  prop.table(margin = 2) %>% 
  as.data.frame() %>% 
  dplyr::rename(Phase = Var1, Sample = Var2) %>% 
  mutate(Phase = factor(Phase, levels = c("G2M", "S", "G1"))) %>%
  ggplot(aes(x = Sample, y = Freq, fill = Phase)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  labs(title = "Alveolar macrophages - Cell cycle phase",
       caption = expression("p = 3.2 x 10"^"-40")) + # OLD: "p = 5.6 x 10"^"-41"
  theme(panel.grid.major.x = element_blank(),
        plot.title = element_text(face = "bold", hjust = 0.5, size = 16),
        axis.title.x = element_blank(),
        axis.text = element_text(color = "black", size = 11),
        plot.caption = element_text(size = 10, hjust = 0.5)) +
  scale_fill_manual(values = c("tomato2", "darkseagreen", "lightskyblue2"))
ggsave("Figures_Flu_pure_RJB/07_AM_DEGs_CellCycle_Bar.pdf")
```

```{r Alveolar macrophages - phagocytosis}
# Fc-gamma-mediated phagocytosis by AMs important for anti-influenza antibody function (He et al., 2017)

Geneset_Phagocytosis_KEGG <- readLines("Gene_sets/KEGG_FC_GAMMA_R_MEDIATED_Phagocytosis.txt")[c(-1, -2)]
Geneset_Phagocytosis_KEGG <- paste0(
 substring(Geneset_Phagocytosis_KEGG, 1, 1),
 tolower(substring(Geneset_Phagocytosis_KEGG, 2)))
Geneset_Phagocytosis_KEGG <- Geneset_Phagocytosis_KEGG[which(Geneset_Phagocytosis_KEGG %in% rownames(Flu_pure))]
Geneset_Phagocytosis_KEGG <- list(Geneset_Phagocytosis_KEGG)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Phagocytosis_KEGG, name = "Geneset_Phagocytosis_KEGG")

Geneset_Phagocytosis_GO <- read_excel("Gene_sets/GO_0060911_Phagocytosis.xlsx") %>% pull(Symbol)
# Already in mouse gene format
Geneset_Phagocytosis_GO <- Geneset_Phagocytosis_GO[which(Geneset_Phagocytosis_GO %in% rownames(Flu_pure))]
Geneset_Phagocytosis_GO <- list(Geneset_Phagocytosis_GO)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Phagocytosis_GO, name = "Geneset_Phagocytosis_GO")


VlnPlot(subset(Flu_pure, idents = c("Alveolar macrophages")), features = c("Geneset_Phagocytosis_KEGG1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.275, 0.4, 0.4)) +
  ylim(-0.1, 0.45) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("KEGG FCgR-mediated phagocytosis geneset - AMs")
# No significant differences

VlnPlot(subset(Flu_pure, idents = c("Alveolar macrophages")), features = c("Geneset_Phagocytosis_GO1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.175, 0.125, 0.125)) +
  ylim(-0.075, 0.2) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("GO phagocytosis geneset - Alveolar macrophages")
ggsave("Figures_Flu_pure_RJB/07_AM_phagocytosis_module_VlnPlot.pdf")
```

```{r Interstitial macrophages - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_IntMac_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Interstitial macrophages")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_IntMac_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)


DotPlot(subset(Flu_pure, idents = c("Interstitial macrophages")), features = c("Cd74", "H2-Ab1", "Nfkb1", "Lilr4b", "Retnlg", "Ccl4", "Cxcl2", "Cxcl10", "Ccl3", "Il1b", "Fabp4", "Hspa1a", "Hspa1b", "Cebpb", "Klf2", "Fabp5", "Clec10a"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Cxcl2 - D3 Y=O, D9 Old
# Cd74 - D0 O > Y; D0 > D3/9
# Fabp5 - D9 O

VlnPlot(subset(Flu_pure, idents = c("Interstitial macrophages")), features = c("Cd86"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Macrophages - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_Mac_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Macrophages")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_Mac_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)


DotPlot(subset(Flu_pure, idents = c("Macrophages")), features = c("Ccrl2", "Cxcl10", "Ccl4", "Cxcl2", "Il1b", "Clec4e", "Ccl3", "Areg", "Hspa1a", "Cxcr4", "Klf2", "Fos", "Apoe"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Cxcl2 - D3 Y=O, D9 Old
# Cxcr4 - D3 Y >~ O, D9 O > Y
# Il1b - D3 O >> Y, D9 O > Y
# Fos - Do O >~ Y, D3 O = Y, D9 O > Y
# Klf2 - D3/9 O > Y
# Apoe - D0 O > Y, D3 O = Y, D9 O > Y


DotPlot(subset(Flu_pure, idents = c("Macrophages")), features = c("Cd80", "Cd86"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(subset(Flu_pure, idents = c("Macrophages")), features = c("Apoe"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Monocytes - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_Mono_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Monocytes")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_Mono_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)


DotPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Ccrl2", "Cxcl10", "Ccl4", "Cxcl2", "Il1b", "Clec4e", "Ccl3", "Areg", "Hspa1a", "Cxcr4", "Klf2", "Fos", "Apoe", "Dnajb1", "Polr2l"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Cxcl2 - D3 Y=O, D9 Old
# Cxcr4 - D3 Y >~ O, D9 O > Y
# Il1b - D3 O >> Y, D9 O > Y
# Fos - Do O >~ Y, D3 O = Y, D9 O > Y
# Klf2 - D3/9 O > Y
# Apoe - D0/3/9 O > Y


DotPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Cd80", "Cd86"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Apoe"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Monocytes - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_DCs_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("cDCs")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_DCs_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)


DotPlot(subset(Flu_pure, idents = c("cDCs")), features = c("Cxcl15", "Ccl17", "Malt1", "Tnfaip3", "Cxcl2", "Cxcl10", "Ccr9", "Rel", "Ccl4"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))




DotPlot(subset(Flu_pure, idents = c("cDCs")), features = c("Cd80", "Cd86"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(subset(Flu_pure, idents = c("cDCs")), features = c("Apoe"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Monocyte to AM transition}
# During lung injury, monocytes can differentiate into AMs (Mo-AMs, distinct from fetal-derived tissue-resident AMs)
FeaturePlot(subset(Flu_pure, idents = "Monocytes"), features = c("Pparg", "Rxra", "Apoa1bp", "Csf2ra"))

# Genes upregulated during Monocytes-AM transition (Misharin et al., 10.1084/jem.20162152)
VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Pparg", "Rxra", "Apoa1bp", "Csf2ra"), group.by = "Timepoint_Age", pt.size = 0, ncol = 2)

# Genes downregulated during Monocytes-AM transition (Misharin et al., 10.1084/jem.20162152)
VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Junb", "Cx3cr1", "Fcer1g", "Treml2"), group.by = "Timepoint_Age", pt.size = 0, ncol = 2)


VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Csf2ra"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4, 4, 4)) +
  ylim(-0.1, 4.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Csf2ra - Monocytes")

VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Junb"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4, 4, 4)) +
  ylim(-0.1, 4.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Junb - Monocytes")

VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Cx3cr1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4, 4, 4)) +
  ylim(-0.1, 4.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Cx3cr1 - Monocytes")

VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Fcer1g"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4, 4, 4)) +
  ylim(-0.1, 4.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Fcer1g - Monocytes")

# Use module scores instead
# K assignment = gene cluster; cluster 1 is upregulated during Mono to AM, cluster 2 is down
Misharin_MoAM_DEGs <- read_excel("Gene_sets/Misharin_Mono_AM_tables/JEM_20162152_TableS2.xlsx")[, 1:2]
# Need to convert Ensembl gene ID to gene name

mart <- useMart(biomart = "ensembl", dataset = "mmusculus_gene_ensembl")
symb <- getBM(attributes = c("ensembl_gene_id","mgi_symbol"),
             filters = "ensembl_gene_id", values = Misharin_MoAM_DEGs$Symbol,
             mart = mart)
Misharin_MoAM_DEGs$gene <- symb$mgi_symbol[match(Misharin_MoAM_DEGs$Symbol, symb$ensembl_gene_id, nomatch = NA)]

# Add two module scores
Geneset_MoAm_Up <- Misharin_MoAM_DEGs %>%
  dplyr::filter(`K-assignment` == 1) %>% 
  pull(gene)
Geneset_MoAm_Up <- Geneset_MoAm_Up[which(Geneset_MoAm_Up %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_MoAm_Up <- list(Geneset_MoAm_Up)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_MoAm_Up, name = "Geneset_MoAm_Up")

Geneset_MoAm_Down <- Misharin_MoAM_DEGs %>%
  dplyr::filter(`K-assignment` == 2) %>% 
  pull(gene)
Geneset_MoAm_Down <- Geneset_MoAm_Down[which(Geneset_MoAm_Down %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_MoAm_Down <- list(Geneset_MoAm_Down)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_MoAm_Down, name = "Geneset_MoAm_Down")


VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Geneset_MoAm_Up1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.045, 0.05, 0.055)) +
  ylim(-0.025, 0.1) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Monocyte to AM upregulated gene set - Monocytes")
ggsave("Figures_Flu_pure_RJB/06_Monocytes_AM_Up_module_VlnPlot.pdf")

VlnPlot(subset(Flu_pure, idents = c("Monocytes")), features = c("Geneset_MoAm_Down1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.375, 0.375, 0.375)) +
  ylim(0, 0.425) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Monocyte to AM downregulated gene set - Monocytes")
ggsave("Figures_Flu_pure_RJB/06_Monocytes_AM_Down_module_VlnPlot.pdf")
```

```{r Monocyte to AM transition - Monocle setup}
# Use Monocle on Monocyte + AM subsetted data, split by age and timepoint

Flu_MoAM <- subset(Flu_pure, idents = c("Monocytes", "Alveolar macrophages"))
dim(Flu_MoAM) # 6092 cells

# Make a CDS Monocle object
### Note that we need to use count data, not scaled data, for Monocle
data_monocle <- newCellDataSet(cellData = as.matrix(Flu_MoAM@assays$RNA@counts), phenoData = new("AnnotatedDataFrame", data = Flu_MoAM@meta.data))
data_monocle <- estimateSizeFactors(data_monocle)
data_monocle <- estimateDispersions(data_monocle)
data_monocle <- detectGenes(data_monocle, min_expr = 0.1)
expressed_genes <- row.names(subset(fData(data_monocle), num_cells_expressed >= 0.05 * ncol(data_monocle)))

# Find the top markers per cluster for Monocle to use
DefaultAssay(Flu_MoAM) <- "RNA"
markers <- FindAllMarkers(Flu_MoAM)
markers <- subset(markers, p_val_adj < 0.05 & abs(avg_log2FC) > 0.25)

# Fewer genes make graph simpler, more genes make graph more complex
top_markers_per_cluster <- markers %>% group_by(cluster) %>% top_n(-100, p_val_adj)
# top_markers_per_cluster <- markers %>% group_by(cluster) %>% top_n(-200, p_val_adj)

data_ordering_genes <- unique(top_markers_per_cluster$gene)

# Run monocle, regressing out mouse sample and the number of genes detected per cell
data_monocle <- setOrderingFilter(data_monocle, ordering_genes = data_ordering_genes)
plot_ordering_genes(data_monocle)
data_monocle <- reduceDimension(data_monocle, reduction_method = "DDRTree", max_components = 2, residualModelFormulaStr = "~orig.ident + num_genes_expressed")
data_monocle <- orderCells(data_monocle, reverse = T)
# Using reverse = T because by default 100 genes only gives one monocle state with AMs set at pseudotime = 0
```

```{r Monocyte to AM transition - Monocle Save/Load}
# saveRDS(data_monocle, "RDS_files.nosync/Flu_pure_Monocyte_AM_Monocle.RDS")
data_monocle <- readRDS("RDS_files.nosync/Flu_pure_Monocyte_AM_Monocle_RJB.RDS")
```

```{r Monocyte to AM transition - Monocle graphs}
plot_cell_trajectory(data_monocle, show_state_number = F, cell_size = 0.5) +
  labs(color = "Monocle state")

gg_color_hue <- function(n, hue_min = 10, hue_max = 280, l = 62, c = 100) {
  hues = seq(hue_min, hue_max, length=n+1)
  hcl(h=hues, l=l, c=c)[1:n]
}

plot_cell_trajectory(data_monocle, color_by = "Named_clusters", cell_size = 0.5) +
  scale_color_manual(name = "Seurat cluster", values = gg_color_hue(35)[c(2, 13)])
ggsave("Figures_Flu_pure_RJB/06_Monocytes_Monocle_SeuratClusters.png")

plot_cell_trajectory(data_monocle, color_by = "Pseudotime", cell_size = 0.5)
ggsave("Figures_Flu_pure_RJB/06_Monocytes_Monocle_Pseudotime.png")
# Already in correct order

plot_cell_trajectory(data_monocle, color_by = "Named_clusters", cell_size = 0.1) +
  scale_color_manual(name = "Seurat cluster", values = gg_color_hue(35)[c(2, 13)]) +
  facet_wrap("Timepoint_Age")
ggsave("Figures_Flu_pure_RJB/06_Monocytes_Monocle_SeuratClusters_Split_TimeAge.png")

dplyr::filter(data_monocle@phenoData@data, Named_clusters == "Monocytes") %>%
  dplyr::group_by(Timepoint_Age) %>% 
  summarise_at(vars(Pseudotime), list(Pseudotime = mean))
# Yes, Aged Monocytes at D3 and D9 are further in pseudotime (closer to AM fate)

dplyr::filter(data_monocle@phenoData@data, Named_clusters == "Monocytes") %>%
  ggplot(aes(x = Timepoint_Age, y = Pseudotime, fill = Timepoint_Age)) +
  geom_violin() +
  theme_minimal() +
  scale_fill_manual(values = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = "Monocyte-AM transition - Monocytes") +
  theme(legend.position = "none",
        panel.grid = element_blank(),
        axis.title.x = element_blank(),
        plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        axis.ticks = element_line(),
        axis.line = element_line(),
        axis.text = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(8, 13, 13)) + ylim(-0, 20)
ggsave("Figures_Flu_pure_RJB/06_Monocytes_Monocle_Pseudotime_Vln.pdf")
### XYZ - Add labels to y-axis in figure ("Monocyte-like" and "AM-like")
```

```{r CD8 T cells - Effector function}
VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"), features = c("Il7r"), group.by = "Timepoint_Age", pt.size = 0)


# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_CD8_TimeAge <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("CD8 T cells") & Timepoint == "Day 9"), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_CD8_TimeAge %>% group_by(cluster) %>% top_n(40, wt = avg_log2FC)

# BACH2 regulates CD8+ T cell differentiation by controlling access of AP-1 factors to enhancers (10.1038/ni.3441)


DotPlot(subset(Flu_pure, idents = c("CD4 T cells"), Timepoint == "Day 9"),
        features = c("Nkg7", "Gzmk", "Lef1", "Ccr7"),
        group.by = "Timepoint_Age")

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Nkg7"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5)) +
  ylim(-0.05, 5.25) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Nkg7_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Gzmk"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(3.5)) +
  ylim(-0.05, 4) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Gzmk_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Lef1"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4)) +
  ylim(-0.05, 4.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Lef1_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Ccr7"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4)) +
  ylim(-0.05, 4.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Ccr7_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Bach2"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(2.75)) +
  ylim(-0.05, 3) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Bach2_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Tcf7"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(3.75)) +
  ylim(-0.05, 4) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Tcf7_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Foxo1"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(2.5)) +
  ylim(-0.05, 2.75) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Foxo1_Vln.pdf", height = 4, width = 4.5)
```

```{r CD8 T cells - SLEC and MPEC module scores}
# Don't use LCMV effector geneset for flu infection, markers may not match up
Geneset_SLEC <- read_tsv("Gene_sets/GSE41978_KLRG1_HIGH_VS_LOW_EFFECTOR_CD8_TCELL_UP.v7.5.1.tsv") %>% 
  dplyr::filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_SLEC <- paste0(
 substring(Geneset_SLEC, 1, 1),
 tolower(substring(Geneset_SLEC, 2))) # Change capitalization
Geneset_SLEC <- Geneset_SLEC[which(Geneset_SLEC %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_SLEC <- list(Geneset_SLEC)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_SLEC, name = "Geneset_SLEC")

Geneset_MPEC <- read_tsv("Gene_sets/GSE41978_KLRG1_HIGH_VS_LOW_EFFECTOR_CD8_TCELL_DN.v7.5.1.tsv") %>% 
  dplyr::filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_MPEC <- paste0(
 substring(Geneset_MPEC, 1, 1),
 tolower(substring(Geneset_MPEC, 2))) # Change capitalization
Geneset_MPEC <- Geneset_MPEC[which(Geneset_MPEC %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_MPEC <- list(Geneset_MPEC)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_MPEC, name = "Geneset_MPEC")

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Geneset_SLEC1"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.085)) +
  ylim(-0.125, 0.095) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = "TE gene set - CD8 T cells")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Module_SLEC_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Geneset_MPEC1"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(1.2)) +
  ylim(0.2, 1.25) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = "MPC gene set - CD8 T cells")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Module_MPEC_Vln.pdf", height = 4, width = 4.5)
```

```{r CD8 T cells - signaling/cytokines}
# Looking at expression based on CellChat predictions
VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Il4ra"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(2.75)) +
  ylim(-0.05, 3) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = "Il4ra - CD8 T cells")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Il4ra_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Cxcr4"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(3.5)) +
  ylim(-0.05, 4) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = "Cxcr4 - CD8 T cells")

VlnPlot(subset(Flu_pure, idents = c("CD8 T cells"), Timepoint == "Day 9"),
        features = c("Ccr5"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(3.75)) +
  ylim(-0.05, 4) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = "Ccr5 - CD8 T cells")
ggsave("Figures_Flu_pure_RJB/08_CD8_D9_Ccr5_Vln.pdf", height = 4, width = 4.5)
```

```{r CD4 T cell DEGs}
Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_CD4_TimeAge <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("CD4 T cells") & Timepoint == "Day 9"), only.pos = T)
Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_CD4_TimeAge %>% group_by(cluster) %>% top_n(50, wt = avg_log2FC)

DotPlot(subset(Flu_pure, idents = c("CD4 T cells"), Timepoint == "Day 9"),
        features = c("Ccr7", "Sell", "Ctla4", "Bhlhe40", "Id2", "Klf3", "Bcl2a1b"),
        group.by = "Timepoint_Age")

# Cite Ryan's paper for Id2 and Klf3

# Bcl2a1b up in Aged - check apoptosis scores for CD4 T cells
# Increased Ctla4 in Aged - Tregs?

# Increased Th1 markers in Aged, increased MPC markers in Young

# Could be kinetic - young have already passed peak cytotxicity, aged are currently at peak

# Tregs?
DotPlot(subset(Flu_pure, idents = c("CD4 T cells"), Timepoint == "Day 9"),
        features = c("Foxp3"),
        group.by = "Timepoint_Age")

VlnPlot(subset(Flu_pure, idents = c("CD4 T cells"), Timepoint == "Day 9"),
        features = c("Ccr7"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5)) +
  ylim(-0.05, 5.25) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD4_D9_Ccr7_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD4 T cells"), Timepoint == "Day 9"),
        features = c("Klf3"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(3.5)) +
  ylim(-0.05, 4) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD4_D9_Klf3_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD4 T cells"), Timepoint == "Day 9"),
        features = c("Id2"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4)) +
  ylim(-0.05, 4.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD4_D9_Id2_Vln.pdf", height = 4, width = 4.5)

VlnPlot(subset(Flu_pure, idents = c("CD4 T cells"), Timepoint == "Day 9"),
        features = c("Bhlhe40"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4)) +
  ylim(-0.05, 4.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_CD4_D9_Bhlhe40_Vln.pdf", height = 4, width = 4.5)
```

```{r CD4 - Th1 module score}
Geneset_Th1 <- readLines("Gene_sets/Th1_Ciucci_2019.txt")
Geneset_Th1 <- paste0(
 substring(Geneset_Th1, 1, 1),
 tolower(substring(Geneset_Th1, 2))) # Change capitalization
Geneset_Th1 <- Geneset_Th1[which(Geneset_Th1 %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_Th1 <- list(Geneset_Th1)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Th1, name = "Geneset_Th1")

# Geneset from Ciucci et al. 2019 (10.1016/j.immuni.2018.12.019)

VlnPlot(subset(Flu_pure, idents = c("CD4 T cells"), Timepoint == "Day 9"),
        features = c("Geneset_Th11"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(1)) +
  ylim(-0.1, 1.1) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = "Th1 gene set - CD4 T cells")
ggsave("Figures_Flu_pure_RJB/08_CD4_D9_Module_Th1_Vln.pdf", height = 4, width = 4.5)
```

```{r T cell Subsets}

Flu_Tcells <- subset(Flu_pure, idents = c("CD8 T cells", "CD4 T cells"))



```



















```{r B cells - DEGs}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_NaiveBcells_TimeAge <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("Naive B cells") & Timepoint == "Day 9"), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_NaiveBcells_TimeAge %>% group_by(cluster) %>% top_n(30, wt = avg_log2FC)

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_PlasmaCells_TimeAge <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("Plasma cells") & Timepoint == "Day 9"), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_PlasmaCells_TimeAge %>% group_by(cluster) %>% top_n(30, wt = avg_log2FC)

VlnPlot(subset(Flu_pure, idents = c("Plasma cells"), Timepoint == "Day 9"),
        features = c("Apoe"),
        group.by = "Timepoint_Age",
        pt.size = 0, cols = c("orange", "orangered3")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank(),
        legend.position = "none") +
  stat_compare_means(
    comparisons = list(c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5)) +
  ylim(-0.05, 5.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

# Flu_pure@active.ident <- Flu_pure$Cluster_Time_Age
# Flu_pure_DEGs_BCells_PlasmaCells_ClusterTimeAge <- FindAllMarkers(subset(Flu_pure, Named_clusters %in% c("Naive B cells", "Plasma cells") & Timepoint == "Day 9"), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_BCells_PlasmaCells_ClusterTimeAge %>% group_by(cluster) %>% top_n(30, wt = avg_log2FC)

DotPlot(subset(Flu_pure, idents = c("Naive B cells", "Plasma cells"), Timepoint == "Day 9"),
        features = c("Cr2", "Btla", "Nr4a1", "Cxcr4", "Fos", "Irf8", "Bach2", "Irf4", "Anxa2", "Plac8", "S100a6", "Ahnak", "Sox5", "Zbtb32", "Itgb2", "Bhlhe41", "Ptpn22", "Prdm1"),
        group.by = "Cluster_Time_Age") +
  theme(axis.title = element_blank()) +
  scale_y_discrete(labels = c(
    paste0(rep("Naive B cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[5:6]),
    paste0(rep("Plasma cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[5:6])
  )) +
  coord_flip()
ggsave("Figures_Flu_pure_RJB/08_Bcell_PlasmaCell_DEGs.pdf")

DotPlot(subset(Flu_pure, idents = c("Naive B cells")),
        features = c("Cr2", "Btla", "Nr4a1", "Cxcr4", "Fos", "Irf8", "Anxa2", "Plac8", "S100a6", "Ahnak", "Sox5", "Zbtb32", "Bhlhe41", "Ptpn22", "Bach2", "Pax5", "Irf4", "Prdm1", "Xbp1"),
        group.by = "Timepoint_Age") +
  coord_flip() +
  theme(axis.title = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1))
ggsave("Figures_Flu_pure_RJB/08_NaiveB_DEGs_DotPlot.pdf")

DotPlot(subset(Flu_pure, idents = c("Plasma cells")),
        features = c("Cr2", "Btla", "Nr4a1", "Cxcr4", "Fos", "Irf8", "Anxa2", "Plac8", "S100a6", "Ahnak", "Sox5", "Zbtb32", "Bhlhe41", "Ptpn22", "Bach2", "Pax5", "Irf4", "Prdm1", "Xbp1"),
        group.by = "Timepoint_Age") +
  coord_flip() +
  theme(axis.title = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1))
ggsave("Figures_Flu_pure_RJB/08_PlasmaCell_DEGs_DotPlot.pdf")

# S100A6 Promotes B Lymphocyte Penetration Through the Blood-Brain Barrier in Autoimmune Encephalitis (10.3389/fgene.2019.01188)
# Irf2bp2 - IRF2-dependent transcriptional repressor (10.1002/JLB.MR1218-507R)
# Ahnak - Marker of B1/B-innate cells in renal transplant rejection (10.1038/s41467-021-24615-6)
# Ptpn22 - Phopsphatase with limited study in B cell types (10.1146/annurev-immunol-032713-120249); no difference in total peripheral B cells (10.1126/science.1092138) or plasma cells (10.4049/jimmunol.0803317) in Ptpn22-/- mice; however, variation in expression here suggests Ptpn22 may play a more important role in B cells with aging; also involved with inhibiting BCR signaling to promote tolerance in T1D
# Bhlhe41 and Bhlhe40 control B1a cell differentiation, but Bhlhe41 reporter was expressed at lo/int levels on splenic plasma cells in naive mice (10.1038/ni.3694); Bhlhe40 restrains the GC reaction in both CD4 T cells and B cells (10.1084/jem.20211406)
# Bach2 - represses differentiation toward plasma cells by inhibiting expression of Blimp1 (Igarashi et al., - 10.1111/imr.12201) - Naive B Y=O, plasma Y = O
# Anxa2 (annexin A2) - Plasma cells D0Y, D3O, D9Y 
# Zbtb32 - repressor of the class II transactivator (CIITA) and MHC class II gene expression during B cell differentiation to plasma cells (10.4049/jimmunol.1103371); also restricts the duration of memory plasma cell responses (10.4049/jimmunol.1600882); may be one mechanism by which aged plasma cells have impaired memory
# IRF8 negatively regulates naive B to plasma cell differentiation in complex with PU.1 (10.1084/jem.20140425)
# IRF4 required for plasma cell differentiation; IRF4 and Blimp1 are both upstream of the TF XBP-1 (10.1038/ni1357)
# Prdm1 (Blimp1) is the master regulator of plasma cell differentiation
# Memory B cells are maintained in aged humans, but antibody response is diminshed; Blimp-1 decreased in aged human B cells in culture (10.1016/j.vaccine.2016.04.023)
# B cell vaccine response decreased in humans with age (j.coi.2014.05.008)
# BCR diversity may not be decreased in mice, at least in response to specific antigens (10.4049/jimmunol.1601106, PMID 6207222)
# Review (10.1111/j.1600-065X.1997.tb01031.x)

VlnPlot(subset(Flu_pure, idents = c("Naive B cells", "Plasma cells"), Timepoint == "Day 3"),
        features = c("Bach2"),
        group.by = "Cluster_Time_Age", pt.size = 0, cols = rep(c("orange", "orangered3"), 2)) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  scale_x_discrete(labels = c(
    paste0(rep("Naive B cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[3:4]),
    paste0(rep("Plasma cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[3:4])
  )) +
  stat_compare_means(
    comparisons = list(
      c("Naive B cells Day 3 Young", "Naive B cells Day 3 Aged"),
      c("Plasma cells Day 3 Young", "Plasma cells Day 3 Aged")),
    label.y = c(4.5, 4.5)) +
  ylim(-0.05, 5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
ggsave("Figures_Flu_pure_RJB/08_Bcell_Bach2_Day3.pdf", width = 6, height = 4)

VlnPlot(subset(Flu_pure, idents = c("Naive B cells", "Plasma cells")),
        features = c("Xbp1"),
        group.by = "Cluster_Time_Age", pt.size = 0,
        cols = rep(c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3"), 2)) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  scale_x_discrete(labels = c(
    paste0(rep("Naive B cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[1:6]),
    paste0(rep("Plasma cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[1:6])
  )) +
  stat_compare_means(
    comparisons = list(
      c("Naive B cells Day 0 Young", "Naive B cells Day 0 Aged"),
      c("Plasma cells Day 0 Young", "Plasma cells Day 0 Aged"),
      c("Naive B cells Day 3 Young", "Naive B cells Day 3 Aged"),
      c("Plasma cells Day 3 Young", "Plasma cells Day 3 Aged"),
      c("Naive B cells Day 9 Young", "Naive B cells Day 9 Aged"),
      c("Plasma cells Day 9 Young", "Plasma cells Day 9 Aged")),
    label.y = rep(4.5, 12)) +
  ylim(-0.05, 5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Naive B cells", "Plasma cells")),
        features = c("Pax5"),
        group.by = "Cluster_Time_Age", pt.size = 0,
        cols = rep(c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3"), 2)) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  scale_x_discrete(labels = c(
    paste0(rep("Naive B cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[1:6]),
    paste0(rep("Plasma cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[1:6])
  )) +
  stat_compare_means(
    comparisons = list(
      c("Naive B cells Day 0 Young", "Naive B cells Day 0 Aged"),
      c("Plasma cells Day 0 Young", "Plasma cells Day 0 Aged"),
      c("Naive B cells Day 3 Young", "Naive B cells Day 3 Aged"),
      c("Plasma cells Day 3 Young", "Plasma cells Day 3 Aged"),
      c("Naive B cells Day 9 Young", "Naive B cells Day 9 Aged"),
      c("Plasma cells Day 9 Young", "Plasma cells Day 9 Aged")),
    label.y = rep(4.5, 12)) +
  ylim(-0.05, 5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Naive B cells", "Plasma cells")),
        features = c("Foxp1"),
        group.by = "Cluster_Time_Age", pt.size = 0,
        cols = rep(c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3"), 2)) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  scale_x_discrete(labels = c(
    paste0(rep("Naive B cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[1:6]),
    paste0(rep("Plasma cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[1:6])
  )) +
  stat_compare_means(
    comparisons = list(
      c("Naive B cells Day 0 Young", "Naive B cells Day 0 Aged"),
      c("Plasma cells Day 0 Young", "Plasma cells Day 0 Aged"),
      c("Naive B cells Day 3 Young", "Naive B cells Day 3 Aged"),
      c("Plasma cells Day 3 Young", "Plasma cells Day 3 Aged"),
      c("Naive B cells Day 9 Young", "Naive B cells Day 9 Aged"),
      c("Plasma cells Day 9 Young", "Plasma cells Day 9 Aged")),
    label.y = rep(4.5, 12)) +
  ylim(-0.05, 5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r B/plasma cell modules}
# Geneset_PlasmaCell <- read_tsv("Gene_sets/TARTE_PLASMA_CELL_VS_B_LYMPHOCYTE_UP.v7.5.1.tsv") %>% # Using the other gene set for consistency (so both memory and plasma gene sets come from the same paper)
Geneset_PlasmaCell <- read_tsv("Gene_sets/GSE13411_PLASMA_CELL_VS_MEMORY_BCELL_UP.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_PlasmaCell <- paste0(
 substring(Geneset_PlasmaCell, 1, 1),
 tolower(substring(Geneset_PlasmaCell, 2))) # Change capitalization
Geneset_PlasmaCell <- Geneset_PlasmaCell[which(Geneset_PlasmaCell %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_PlasmaCell <- list(Geneset_PlasmaCell)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_PlasmaCell, name = "Geneset_PlasmaCell")

Geneset_MemoryB_Naive <- read_tsv("Gene_sets/GSE13411_NAIVE_VS_MEMORY_BCELL_DN.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_MemoryB_Naive <- paste0(
 substring(Geneset_MemoryB_Naive, 1, 1),
 tolower(substring(Geneset_MemoryB_Naive, 2))) # Change capitalization
Geneset_MemoryB_Naive <- Geneset_MemoryB_Naive[which(Geneset_MemoryB_Naive %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_MemoryB_Naive <- list(Geneset_MemoryB_Naive)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_MemoryB_Naive, name = "Geneset_MemoryB_Naive")

Geneset_MemoryB_Plasma <- read_tsv("Gene_sets/GSE13411_PLASMA_CELL_VS_MEMORY_BCELL_DN.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_MemoryB_Plasma <- paste0(
 substring(Geneset_MemoryB_Plasma, 1, 1),
 tolower(substring(Geneset_MemoryB_Plasma, 2))) # Change capitalization
Geneset_MemoryB_Plasma <- Geneset_MemoryB_Plasma[which(Geneset_MemoryB_Plasma %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_MemoryB_Plasma <- list(Geneset_MemoryB_Plasma)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_MemoryB_Plasma, name = "Geneset_MemoryB_Plasma")


VlnPlot(subset(Flu_pure, idents = c("Naive B cells", "Plasma cells")),
        features = c("Geneset_PlasmaCell1"),
        group.by = "Cluster_Time_Age", pt.size = 0,
        cols = rep(c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3"), 2)) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  scale_x_discrete(labels = c(
    paste0(rep("Naive B cells", 6), "\n", levels(Flu_pure$Timepoint_Age)[1:6]),
    paste0(rep("Plasma cells", 6), "\n", levels(Flu_pure$Timepoint_Age)[1:6])
  )) +
  stat_compare_means(
    comparisons = list(
      c("Naive B cells Day 0 Young", "Naive B cells Day 0 Aged"),
      c("Naive B cells Day 3 Young", "Naive B cells Day 3 Aged"),
      c("Naive B cells Day 9 Young", "Naive B cells Day 9 Aged"),
      c("Plasma cells Day 0 Young", "Plasma cells Day 0 Aged"),
      c("Plasma cells Day 3 Young", "Plasma cells Day 3 Aged"),
      c("Plasma cells Day 9 Young", "Plasma cells Day 9 Aged")),
    label.y = c(0.2, 0.15, 0.15, 0.26, 0.15, 0.15)) +
  ylim(-0.12, 0.3) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = c("Plasma cell gene set"))
ggsave("Figures_Flu_pure_RJB/08_PlasmaCell_Module.pdf", height = 4, width = 14)


VlnPlot(subset(Flu_pure, idents = c("Naive B cells", "Plasma cells")),
        features = c("Geneset_MemoryB_Plasma1"),
        group.by = "Cluster_Time_Age", pt.size = 0,
        cols = rep(c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3"), 2)) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  scale_x_discrete(labels = c(
    paste0(rep("Naive B cells", 6), "\n", levels(Flu_pure$Timepoint_Age)[1:6]),
    paste0(rep("Plasma cells", 6), "\n", levels(Flu_pure$Timepoint_Age)[1:6])
  )) +
  stat_compare_means(
    comparisons = list(
      c("Naive B cells Day 0 Young", "Naive B cells Day 0 Aged"),
      c("Naive B cells Day 3 Young", "Naive B cells Day 3 Aged"),
      c("Naive B cells Day 9 Young", "Naive B cells Day 9 Aged"),
      c("Plasma cells Day 0 Young", "Plasma cells Day 0 Aged"),
      c("Plasma cells Day 3 Young", "Plasma cells Day 3 Aged"),
      c("Plasma cells Day 9 Young", "Plasma cells Day 9 Aged")),
    label.y = rep(0.55, 12)) +
  ylim(-0.1, 0.6) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = c("Memory B cell gene set"))
ggsave("Figures_Flu_pure_RJB/08_MemoryB_Module.pdf", height = 4, width = 14)




VlnPlot(subset(Flu_pure, idents = c("Naive B cells", "Plasma cells"), Timepoint == "Day 9"),
        features = c("Geneset_MemoryB_Naive1"),
        group.by = "Cluster_Time_Age", pt.size = 0, cols = rep(c("orange", "orangered3"), 2)) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  scale_x_discrete(labels = c(
    paste0(rep("Naive B cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[5:6]),
    paste0(rep("Plasma cells", 2), "\n", levels(Flu_pure$Timepoint_Age)[5:6])
  )) +
  stat_compare_means(
    comparisons = list(
      c("Naive B cells Day 9 Young", "Naive B cells Day 9 Aged"),
      c("Plasma cells Day 9 Young", "Plasma cells Day 9 Aged")),
    label.y = c(0.35, 0.35)) +
  ylim(-0.2, 0.4) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = c("Memory B cell vs naive gene set"))
```

```{r Endothelial cells - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_EC_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Endothelial cells")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_EC_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)


DotPlot(subset(Flu_pure, idents = c("Endothelial cells")), features = c("Cxcl10", "Irf1", "Nfkbiz", "Icam1", "Egr1", "Ahr", "Sox18", "Clec2d", "Adgre5", "Efna1", "Hspa1b", "Aplnr", "Vwf", "Id1", "Klf2", "Foxf1"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Cxcl10 - D3 O > Y
# Irf1 - D3 Y = O, D9 O > Y; D3 > D9
# Icam1 - D3 O > Y, D9 O > Y; D3 > D9
# Ahr - D3 Y > O, D9 Y > O; D0 < D3 < D9
# Sox18 (TF that cooperates with Sox7 and Sox17 to promote vascular development and proliferation (10.1371/journal.pone.0143650)) - D0 Y = O, D3/9 Y > O; D0 < D3 < D9
#     Could also use SCENIC for these 3 Sox TFs
# Clec2d (ligand for Klrb1, senses cell death by recognizing histones but usually expressed on macrophages (10.1016/j.immuni.2019.11.013)) - D0 Y = O, D3/9 Y > O; D0 < D3 < D9
# Vwf - D0/3/9 O > Y
# Klf2 - D3 O > Y, D9 O > Y; D9 > D3
# Foxf1 (TF involved in vascular sprouting, upregulated in progenitor-like ECs compared to mature ECs, upregulates Notch2, Vegfr2 (Kdr), and ephrin B2 (Efnb2) but downregulates Ephb4 (10.3389/fbioe.2018.00076)) - D0 O > Y, D3 Y > O, D9 O > Y; D0 < D3 < D9
#   Kdr downregulated in O vs Y on D9 even though Foxf1 is higher in Old; impaired Foxf1 function?
#   Could try running SCENIC on Endothelial cells split by Timepoint_Age


DotPlot(subset(Flu_pure, idents = c("Endothelial cells")), features = c("Foxf1", "Notch2", "Kdr", "Efnb2", "Ephb4"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

DotPlot(subset(Flu_pure, idents = c("Endothelial cells")), features = c("Sox18", "Sox7", "Sox17"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")), features = c("Sox18"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")



VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")), features = c("S.Score"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.2, 0.3) + 
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")), features = c("G2M.Score"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.5, 0.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")), features = c("Klf2"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.5, 7) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Car4+ endothelial cells - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_Car4EC_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Car4+ endothelial cells")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_Car4EC_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)


DotPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")), features = c("Sdpr", "Cd97", "Egr1", "Egr3", "Hspa1a", "Ier3", "Irf1", "Icam1", "Ahr", "Plvap", "Clec2d", "Gsn", "Inmt", "Mgp", "Id1", "Klf2", "Hilpda", "Hif1a", "Vegfa"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title = element_blank()) +
  coord_flip()




DotPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")), features = c("Foxf1", "Notch2", "Kdr", "Efnb2", "Ephb4"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")), features = c("Clec2d"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")), features = c("Klf2"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 7.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
# KLF2 prevents coagulation and upregulates eNOS to decrease inflammation
# 10.1084/jem.20031132 and 10.1161/01.res.0000159707.05637.a1
# Interesting that Klf2 is strongly up in Aged compared to Young at Day 9 in Car4+ ECs
# May be reactionary to increased levels of inflammation in Aged

# Icam1 up in aged in both regular and Car4+ ECs
VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")), features = c("Icam1"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 7.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")), features = c("Vegfa"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 7.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Car4 EC proliferation}
VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("S.Score"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.5, 0.5, 0.5)) +
  ylim(-0.2, 0.6) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("G2M.Score"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.15, 0.15, 0.15)) +
  ylim(-0.3, 0.2) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")


# (table(subset(Flu_pure, idents = "Car4+ endothelial cells")$Phase,
#       subset(Flu_pure, idents = "Car4+ endothelial cells")$Timepoint_Age) %>% 
#   as.matrix() %>% 
#   chisq.test(x = .))$p.value
# p = 1.524476e-21

table(subset(Flu_pure, idents = "Car4+ endothelial cells")$Phase,
      subset(Flu_pure, idents = "Car4+ endothelial cells")$Timepoint_Age) %>% 
  prop.table(margin = 2) %>% 
  as.data.frame() %>% 
  dplyr::rename(Phase = Var1, Sample = Var2) %>% 
  mutate(Phase = factor(Phase, levels = c("G2M", "S", "G1"))) %>%
  ggplot(aes(x = Sample, y = Freq, fill = Phase)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  labs(title = "Car4+ ECs - Cell cycle phase",
       caption = expression("p = 1.5 x 10"^"-21")) +
  theme(panel.grid.major.x = element_blank(),
        plot.title = element_text(face = "bold", hjust = 0.5, size = 16),
        axis.title.x = element_blank(),
        axis.text = element_text(color = "black", size = 11),
        plot.caption = element_text(size = 10, hjust = 0.5)) +
  scale_fill_manual(values = c("tomato2", "darkseagreen", "lightskyblue2"))


# (table(subset(Flu_pure, idents = "Endothelial cells")$Phase,
#       subset(Flu_pure, idents = "Endothelial cells")$Timepoint_Age) %>% 
#   as.matrix() %>% 
#   chisq.test(x = .))$p.value
# p = 4.346226e-10

table(subset(Flu_pure, idents = "Endothelial cells")$Phase,
      subset(Flu_pure, idents = "Endothelial cells")$Timepoint_Age) %>% 
  prop.table(margin = 2) %>% 
  as.data.frame() %>% 
  dplyr::rename(Phase = Var1, Sample = Var2) %>% 
  mutate(Phase = factor(Phase, levels = c("G2M", "S", "G1"))) %>%
  ggplot(aes(x = Sample, y = Freq, fill = Phase)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  labs(title = "ECs - Cell cycle phase",
       caption = expression("p = 4.3 x 10"^"-10")) +
  theme(panel.grid.major.x = element_blank(),
        plot.title = element_text(face = "bold", hjust = 0.5, size = 16),
        axis.title.x = element_blank(),
        axis.text = element_text(color = "black", size = 11),
        plot.caption = element_text(size = 10, hjust = 0.5)) +
  scale_fill_manual(values = c("tomato2", "darkseagreen", "lightskyblue2"))
```

```{r Car4 EC inflammation modules}
VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Geneset_Inflammation1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.2, 0.25, 0.2)) +
  ylim(-0.15, 0.3) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  labs(title = expression(bold("Hallmark Inflammatory Response gene set - Car4" ^ "+" ~ "ECs"))) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/12_Car4EC_Inflammation_Module_vln.pdf")


Geneset_EC_IFNg_response <- read_tsv("Gene_sets/GSE3920_UNTREATED_VS_IFNG_TREATED_ENDOTHELIAL_CELL_DN.v7.5.1.tsv") %>%
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_EC_IFNg_response <- paste0(
 substring(Geneset_EC_IFNg_response, 1, 1),
 tolower(substring(Geneset_EC_IFNg_response, 2)))
Geneset_EC_IFNg_response <- Geneset_EC_IFNg_response[which(Geneset_EC_IFNg_response %in% rownames(Flu_pure))]
Geneset_EC_IFNg_response <- list(Geneset_EC_IFNg_response)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_EC_IFNg_response, name = "Geneset_EC_IFNg_response")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Geneset_EC_IFNg_response1"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.2, 0.25, 0.2)) +
  ylim(-0.15, 0.3) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Car4 vs standard EC DEGs}
# Car4_vs_EC_DEGs <- FindMarkers(Flu_pure, ident.1 = "Car4+ endothelial cells", ident.2 = c("Endothelial cells"))
# Car4_vs_EC_DEGs$Gene <- rownames(Car4_vs_EC_DEGs)

DotPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells", "Endothelial cells")),
        features = c("Kit", "Cxcl12", "Itga1", "Car4", "Tbx3", "Kdr", "Ednrb", "Cd34", "Tbx2", "Gdf15", "Klf2", "Id1", "Irf1", "Irf9", "Foxf1", "Pecam1", "Icam1", "Icam2", "Egr1", "Cav1", "Cav2", "Thbd"),
        group.by = "Cluster_Time_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title = element_blank()) +
  scale_y_discrete(labels = c(
    paste0(rep("Endothelial cells", 6), "\n", levels(Flu_pure$Timepoint_Age)),
    paste0(rep("Car4+ endothelial cells", 6), "\n", levels(Flu_pure$Timepoint_Age))
  )) +
  coord_flip()
ggsave("Figures_Flu_pure_RJB/12_EC_Car4EC_DEGs_DotPlot.pdf", width = 11, height = 6)
# c-kit(+) cells adopt vascular endothelial but not epithelial cell fates during lung maintenance and repair (10.1038/nm.3888)
# Downregulation of c-kit expression in human endothelial cells by inflammatory stimuli (PMID: 9207448)
# Tbx2 is expressed in embryonic lung mesenchymal progenitors that can give rise to endothelial cells (10.1186/s12931-019-1264-y)
# Ednrb - endothelin receptor B
# Kit (c-kit) - membrane bound receptor on progenitor endothelial cells, activated by KitL (ligand) after inflammation to regenerate vasculature (10.1182/blood-2006-06-029603 - Patrizia et al.)

DotPlot(Flu_pure, features = c("Cxcl12", "Cxcr4"))

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells", "Endothelial cells")),
        features = c("Tbx2", "Tbx3", "Kdr", "Ednrb", "Icam2"),
        split.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3"))

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Vwf"),
        group.by = "Timepoint_Age", pt.size = 0.1,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.5, 4.5, 4.5)) +
  ylim(-0.05, 5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Vwf - Car4+ endothelial cells")

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Vwf"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.75, 4.75, 4.75)) +
  ylim(-0.05, 5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Vwf - Endothelial cells") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Thbd"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.75, 4.75, 4.75)) +
  ylim(-0.05, 5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Thbd - Endothelial cells") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))
```

```{r Platelet DEGs}
# Platelet_DEGs <- FindMarkers(Flu_pure, ident.1 = "Platelets", only.pos = T)

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_Platelets_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Platelets")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters

Flu_pure_DEGs_Platelets_TimeAge %>% dplyr::group_by(cluster) %>% top_n(10, wt = avg_log2FC)
```

```{r Endothelial activation module}
Geneset_EC_activation <- read_tsv("Gene_sets/GOBP_ENDOTHELIAL_CELL_ACTIVATION.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_EC_activation <- paste0(
 substring(Geneset_EC_activation, 1, 1),
 tolower(substring(Geneset_EC_activation, 2)))
Geneset_EC_activation <- Geneset_EC_activation[which(Geneset_EC_activation %in% rownames(Flu_pure))]
Geneset_EC_activation <- list(Geneset_EC_activation)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_EC_activation, name = "Geneset_EC_activation")

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Geneset_EC_activation1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(1.1, 1.1, 1.1)) +
  ylim(-0.3, 1.2) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Endothelial activation gene set - ECs") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Geneset_EC_activation1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(1, 1, 1)) +
  ylim(-0.3, 1.1) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Endothelial activation gene set - Car4+ ECs") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))
```

```{r Coagulation pathway modules}
Geneset_CommonPathway <- readLines("Gene_sets/REACTOME_COMMON_PATHWAY_OF_FIBRIN_CLOT_FORMATION.txt")[c(-1, -2)]
Geneset_CommonPathway <- paste0(
 substring(Geneset_CommonPathway, 1, 1),
 tolower(substring(Geneset_CommonPathway, 2)))
Geneset_CommonPathway <- Geneset_CommonPathway[which(Geneset_CommonPathway %in% rownames(Flu_pure))]
Geneset_CommonPathway <- list(Geneset_CommonPathway)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_CommonPathway, name = "Geneset_CommonPathway")

Geneset_Coagulation_GO <- read_tsv("Gene_sets/GOBP_COAGULATION.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_Coagulation_GO <- paste0(
 substring(Geneset_Coagulation_GO, 1, 1),
 tolower(substring(Geneset_Coagulation_GO, 2)))
Geneset_Coagulation_GO <- Geneset_Coagulation_GO[which(Geneset_Coagulation_GO %in% rownames(Flu_pure))]
Geneset_Coagulation_GO <- list(Geneset_Coagulation_GO)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Coagulation_GO, name = "Geneset_Coagulation_GO")

Geneset_Coagulation_Hallmark <- read_tsv("Gene_sets/HALLMARK_COAGULATION.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_Coagulation_Hallmark <- paste0(
 substring(Geneset_Coagulation_Hallmark, 1, 1),
 tolower(substring(Geneset_Coagulation_Hallmark, 2)))
Geneset_Coagulation_Hallmark <- Geneset_Coagulation_Hallmark[which(Geneset_Coagulation_Hallmark %in% rownames(Flu_pure))]
Geneset_Coagulation_Hallmark <- list(Geneset_Coagulation_Hallmark)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Coagulation_Hallmark, name = "Geneset_Coagulation_Hallmark")



VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Geneset_CommonPathway1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.55, 0.55, 0.55)) +
  ylim(-0.15, 0.6) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Common Pathway gene set - Endothelial cells") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/12_EC_Module_CommonCoag_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Geneset_CommonPathway1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.55, 0.55, 0.55)) +
  ylim(-0.15, 0.6) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Common Pathway gene set - Car4+ endothelial cells") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/12_Car4EC_Module_CommonCoag_Vln.pdf")



VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Geneset_Coagulation_GO1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.22, 0.22, 0.22)) +
  ylim(-0.05, 0.25) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("GO Coagulation gene set - Endothelial cells") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/12_EC_Module_Coag_GO_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Geneset_Coagulation_GO1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.27, 0.27, 0.27)) +
  ylim(-0.05, 0.3) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("GO Coagulation gene set - Car4+ endothelial cells") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/12_Car4EC_Module_Coag_GO_Vln.pdf")


VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Geneset_Coagulation_Hallmark1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.24, 0.24, 0.24)) +
  ylim(-0.12, 0.27) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("Hallmark Coagulation gene set - Endothelial cells") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/12_EC_Module_Coag_Hallmark_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Geneset_Coagulation_Hallmark1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.28, 0.28, 0.28)) +
  ylim(-0.08, 0.3) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  theme(legend.position = "none",
        axis.title = element_blank(),
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  labs(title = expression(bold("Hallmark Coagulation gene set - Car4" ^ "+" ~ "ECs")))
ggsave("Figures_Flu_pure_RJB/12_Car4EC_Module_Coag_Hallmark_Vln.pdf")
```

```{r Myofibroblasts - DEGs by Timepoint_Age}
DefaultAssay(Flu_pure) <- "RNA"

# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_Myofib_TimeAge <- FindAllMarkers(subset(Flu_pure, subset = Named_clusters == c("Myofibroblasts")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
Flu_pure_DEGs_Myofib_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)


DotPlot(subset(Flu_pure, idents = c("Myofibroblasts")), features = c("Irf1", "Atf3", "Cxcl10", "Cxcl1", "Col1a1", "Col3a1", "C3", "C4b", "Cxcl12", "Socs3", "Klf2", "Dcn", "Ptgs2"), group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Cxcl10 - D3/9 O > Y
# Cxcl1 - D9 O > Y; D3 > D9
# Cxcl2 - D3 Y=O, D9 Old
# Cxcl12 - D9 Old
# Socs3 - D9 O
# Klf2 - D3 Y >~ O, D9 O > Y
# C3 - D9 O
# C4b - D9 O
# Col1a1 - D3/9 Y > O
# Col3a1 - D0/3/9 Y > O
# Dcn (Decorin; binds to type I collagen, may prevent fibrosis by binding and inhibiting TGF-b (10.1016/j.mehy.2021.110612); conversely, also found to be increased in fibrotic lungs (10.1016/j.biocel.2004.01.009)) - D9 O
# Ptgs2 (cyclooxygenase 2, COX-2) - D9 O


VlnPlot(subset(Flu_pure, idents = c("Myofibroblasts")), features = c("Ptgs2"), group.by = "Timepoint_Age", pt.size = 0) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid")
```

```{r Fibrosis - Wfdc2 and Prss23}
# Visium shows Wfdc2 may promote fibrosis, Prss23 inhibits it
# Both of these have been reported
# Wfdc17 (encodes Amwap) may be novel, seems to promote fibrosis, and is produced by immune cells rather than epithelial cells (especially Fn1+ macrophages)
# Amwap (Wfdc17) promotes anti-inflammatory functions in microglia (reduced IL-1, increased Arg1) (10.4049/jimmunol.0903300)

DotPlot(Flu_pure, features = c("Prss23", "Wfdc2", "Wfdc17", "Cd93"))
# Prss23 mostly expressed in Vwf+ endothelial cells, lymphatic endothelial cells, mesothelial cells, and matrix fibroblasts
# Wfdc2 mostly expressed by lung parenchyma - AT2, goblet, club, ciliated cells
# Wfdc17 mostly expressed by myeloid cells
# Cd93 up in nonfibroic inflamed areas

DotPlot(Flu_pure, features = c("Tgfb1", "Wfdc17", "Prss23", "Ehd2", "Hpgd", "Calcrl", "Cav1", "Cavin2", "Wfdc2")) +
  theme(axis.title = element_blank())
ggsave("Figures_Flu_pure_RJB/13_Fibrosis_genes_DotPlot.pdf", width = 10, height = 6)

DotPlot(Flu_pure, features = c("Prss23", "Wfdc2", "Wfdc17", "Tgfb1", "Hpgd", "Calcrl"), split.by = "Timepoint", cols = c("red", "red", "red"))
# Expression over time, so focus on Day 9

VlnPlot(subset(Flu_pure, idents = c("Matrix fibroblasts", "Vwf+ endothelial cells", "Mesothelial cells", "Lymphatic endothelial cells")),
        features = c("Prss23"), split.by = "Timepoint_Age", pt.size = 0)


VlnPlot(subset(Flu_pure, idents = c("Matrix fibroblasts", "Vwf+ endothelial cells", "Mesothelial cells", "Lymphatic endothelial cells"), Timepoint == "Day 9"),
        features = c("Prss23"), split.by = "Age", pt.size = 0) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 0.75, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("orange", "orangered3"), labels = c("Day 9 Young", "Day 9 Aged"))

VlnPlot(subset(Flu_pure, idents = c("AT2 cells", "Goblet cells", "Ciliated cells", "Club cells"), Timepoint == "Day 9"),
        features = c("Wfdc2"), split.by = "Age", pt.size = 0) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 0.75, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("orange", "orangered3"), labels = c("Day 9 Young", "Day 9 Aged"))

VlnPlot(subset(Flu_pure, idents = c("Neutrophils", "Monocytes", "Suppressive neutrophils", "Fn1+ macrophages", "Interstitial macrophages", "Alveolar macrophages"), Timepoint == "Day 9"),
        features = c("Wfdc17"), split.by = "Age", pt.size = 0) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 0.75, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("orange", "orangered3"), labels = c("Day 9 Young", "Day 9 Aged"))

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells", "Car4+ endothelial cells", "Vwf+ endothelial cells", "Pericytes", "Platelets", "Lymphatic endothelial cells"), Timepoint == "Day 9"),
        features = c("Hpgd"), split.by = "Age", pt.size = 0) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 0.75, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("orange", "orangered3"), labels = c("Day 9 Young", "Day 9 Aged"))

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells", "Car4+ endothelial cells", "Vwf+ endothelial cells", "Pericytes", "Platelets", "Lymphatic endothelial cells"), Timepoint == "Day 9"),
        features = c("Calcrl"), split.by = "Age", pt.size = 0) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 0.75, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("orange", "orangered3"), labels = c("Day 9 Young", "Day 9 Aged"))




VlnPlot(subset(Flu_pure, idents = c("Monocytes")),
        features = c("Wfdc17"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5, 5, 5)) +
  ylim(-0.1, 5.5) +
  labs(title = "Wfdc17 - Monocytes") +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/13_Wfdc17_Monocytes.pdf")

VlnPlot(subset(Flu_pure, idents = c("Interstitial macrophages")),
        features = c("Wfdc17"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5.1, 5.1, 5.1)) +
  ylim(-0.1, 5.5) +
  labs(title = "Wfdc17 - Interstitial macrophages") +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/13_Wfdc17_Interstitial_Macrophages.pdf")

VlnPlot(subset(Flu_pure, idents = c("Macrophages")),
        features = c("Wfdc17"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(6, 6, 6)) +
  ylim(-0.1, 6.5) +
  labs(title = "Wfdc17 - Macrophages") +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/13_Wfdc17_Macrophages.pdf")

VlnPlot(subset(Flu_pure, idents = c("Alveolar macrophages")),
        features = c("Wfdc17"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5.1, 4, 4)) +
  ylim(-0.1, 5.5) +
  labs(title = "Wfdc17 - Alveolar macrophages") +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5))
ggsave("Figures_Flu_pure_RJB/13_Wfdc17_AMs.pdf")

# Do Wfdc17+ myeloid cells (day 3 Y vs O) also express higher Tgfb1? Would suggest that Wfdc17 promotes fibrosis via myeloid skewing towards an anti-inflammatory/MDSC phenotype.
data.frame(
  Sample = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")$Timepoint_Age,
  Wfdc17 = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")@assays$RNA["Wfdc17", ] %>% as.numeric(),
  Tgfb1 = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")@assays$RNA["Tgfb1", ] %>% as.numeric()
) %>% 
  ggplot(aes(x = Wfdc17, y = Tgfb1, color = Sample)) +
  geom_point() +
  geom_smooth(aes(group = Sample, color = Sample), method = "lm", se = F, size = 1.25) +
  xlab("Wfdc17 expression") +
  ylab("Tgfb1 expression") +
  theme_minimal() +
  labs(title = "Monocytes") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_color_manual(values = c("lightskyblue3", "royalblue3"))

summary(lm(Tgfb1 ~ Wfdc17,
           data = filter(
             data.frame(
               Sample = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")$Timepoint_Age,
               Wfdc17 = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")@assays$RNA["Wfdc17", ] %>% as.numeric(),
               Tgfb1 = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")@assays$RNA["Tgfb1", ] %>% as.numeric()), 
             Sample == "Day 3 Young")))

summary(lm(Tgfb1 ~ Wfdc17,
           data = filter(
             data.frame(
               Sample = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")$Timepoint_Age,
               Wfdc17 = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")@assays$RNA["Wfdc17", ] %>% as.numeric(),
               Tgfb1 = subset(Flu_pure, idents = "Monocytes", Timepoint == "Day 3")@assays$RNA["Tgfb1", ] %>% as.numeric()), 
             Sample == "Day 3 Aged")))

  

data.frame(
  Sample = subset(Flu_pure, idents = "Interstitial macrophages", Timepoint == "Day 9")$Timepoint_Age,
  Wfdc17 = subset(Flu_pure, idents = "Interstitial macrophages", Timepoint == "Day 9")@assays$RNA["Wfdc17", ] %>% as.numeric(),
  Tgfb1 = subset(Flu_pure, idents = "Interstitial macrophages", Timepoint == "Day 9")@assays$RNA["Tgfb1", ] %>% as.numeric()
) %>% 
  ggplot(aes(x = Wfdc17, y = Tgfb1, color = Sample)) +
  geom_point() +
  geom_smooth(aes(group = Sample, color = Sample), method = "lm", se = F, size = 1.25) +
  xlab("Wfdc17 expression") +
  ylab("Tgfb1 expression") +
  theme_minimal() +
  labs(title = "Interstitial macrophages") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_color_manual(values = c("orange", "orangered3"))

data.frame(
  Sample = subset(Flu_pure, idents = "Macrophages", Timepoint == "Day 3")$Timepoint_Age,
  Wfdc17 = subset(Flu_pure, idents = "Macrophages", Timepoint == "Day 3")@assays$RNA["Wfdc17", ] %>% as.numeric(),
  Tgfb1 = subset(Flu_pure, idents = "Macrophages", Timepoint == "Day 3")@assays$RNA["Tgfb1", ] %>% as.numeric()
) %>% 
  ggplot(aes(x = Wfdc17, y = Tgfb1, color = Sample)) +
  geom_point() +
  geom_smooth(aes(group = Sample, color = Sample), method = "lm", se = F, size = 1.25) +
  xlab("Wfdc17 expression") +
  ylab("Tgfb1 expression") +
  theme_minimal() +
  labs(title = "Macrophages") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_color_manual(values = c("lightskyblue3", "royalblue3"))

data.frame(
  Sample = subset(Flu_pure, idents = "Alveolar macrophages", Timepoint == "Day 9")$Timepoint_Age,
  Wfdc17 = subset(Flu_pure, idents = "Alveolar macrophages", Timepoint == "Day 9")@assays$RNA["Wfdc17", ] %>% as.numeric(),
  Tgfb1 = subset(Flu_pure, idents = "Alveolar macrophages", Timepoint == "Day 9")@assays$RNA["Tgfb1", ] %>% as.numeric()
) %>% 
  ggplot(aes(x = Wfdc17, y = Tgfb1, color = Sample)) +
  geom_point() +
  geom_smooth(aes(group = Sample, color = Sample), method = "lm", se = F, size = 1.25) +
  xlab("Wfdc17 expression") +
  ylab("Tgfb1 expression") +
  theme_minimal() +
  labs(title = "Alveolar macrophages") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_color_manual(values = c("orange", "orangered3"))




VlnPlot(subset(Flu_pure, idents = c("Matrix fibroblasts")),
        features = c("Prss23"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid", position = position_dodge(width = 0.9)) +
  stat_compare_means(comparisons = list(
    c("Day 0 Young", "Day 0 Aged"),
    c("Day 3 Young", "Day 3 Aged"),
    c("Day 9 Young", "Day 9 Aged"))) +
  ylim(-0.1, 9)
```

```{r Epithelial - Wound healing}
Geneset_wound_healing <- read_excel("Gene_sets/GO_0042060_wound_healing.xlsx") %>% pull(Symbol)
# Already in mouse gene format
Geneset_wound_healing <- Geneset_wound_healing[which(Geneset_wound_healing %in% rownames(Flu_pure))]
Geneset_wound_healing <- list(Geneset_wound_healing)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_wound_healing, name = "Geneset_wound_healing")

VlnPlot(Flu_pure, features = c("Geneset_wound_healing1"), pt.size = 0) +
  NoLegend()
# Epithelial cells (myofibroblasts, matrix fibroblasts, mesothelial) all score highly and would normally be involved in wound healing, so use these clusters

VlnPlot(subset(Flu_pure, idents = c("Myofibroblasts", "Matrix fibroblasts", "Mesothelial cells")),
        features = c("Geneset_wound_healing1"),
        group.by = "Timepoint_Age",
        pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.275, 0.275, 0.275)) +
  ylim(-0.05, 0.3) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  ggtitle("GO wound healing geneset - Epithelial cells")
ggsave("Figures_Flu_pure_RJB/13_Epithelial_wound_healing_module_VlnPlot.pdf")
```

```{r Endothelial - DEGs}
# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_Endo_TimeAge <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("Endothelial cells")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
# Flu_pure_DEGs_Endo_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)

DotPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Cbr2", "Sdpr", "Cd97", "Cxcl10", "Egr1", "Icam1", "Irf1", "Atf3", "Ahr", "Clec2d", "Sox18", "Adgre5", "Klf2", "Foxf1", "Id1", "Vwf"),
        group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        # features = c("Egr1"),
        # features = c("Icam1"),
        # features = c("Irf1"),
        # features = c("Atf3"),
        features = c("Cxcl10"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(6, 6, 6)) +
  ylim(-0.05, 7)

# FOXF1 Mediates Endothelial Progenitor Functions and Regulates Vascular Sprouting (10.3389/fbioe.2018.00076)
# Id1 expression in endothelial cells of the colon is required for normal response to injury (10.1016/j.ajpath.2015.07.005)
# Lentivirus‑mediated overexpression of CD97/ADGRE5 reverses dysregulated high glucose‑induced endothelial cell migration (10.3892/mmr.2017.6417)
# SDPR/Cavin-2 in Endothelial Cells Contributes to Inflammatory Cell Adhesion in Abdominal Aortic Aneurysm (https://www.ahajournals.org/doi/10.1161/circ.140.suppl_1.13378)
# Egr1 - TF upregulated in response to vascular injury, upregulates the apoptosis ligand TRAIL (10.1016/s0378-1119(03)00730-3)
# Icam1 - Required for WBC migration into tissues from vessels
# Irf1 upregulated in response to TNFa-Tnfr2 signaling in endothelial cells (10.1016/j.immuni.2013.01.012)



# Tight junction (Ocln, occludin; Cldn1/5/12, claudins; Jam2) and adherens junction (Nectin1/2/3) genes
DotPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Ocln", "Cldn1", "Cldn5", "Cldn12", "Jam2", "Nectin1", "Nectin2", "Nectin3"),
        group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r Endothelial - Vascular wound healing module}
Geneset_Vascular_healing <- read_tsv("Gene_sets/GOBP_POSITIVE_REGULATION_OF_VASCULAR_WOUND_HEALING.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_Vascular_healing <- paste0(
 substring(Geneset_Vascular_healing, 1, 1),
 tolower(substring(Geneset_Vascular_healing, 2))) # Change capitalization
Geneset_Vascular_healing <- Geneset_Vascular_healing[which(Geneset_Vascular_healing %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_Vascular_healing <- list(Geneset_Vascular_healing)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Vascular_healing, name = "Geneset_Vascular_healing")

VlnPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Geneset_Vascular_healing1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(1.3, 0.8, 1.3)) +
  ylim(-0.6, 1.4) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank()) +
  labs(title = "GO Vascular Wound Healing gene set - Endothelial cells")
ggsave("Figures_Flu_pure_RJB/12_Endothelial_Module_VascWoundHealing_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Geneset_Vascular_healing1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.9, 0.9, 0.9)) +
  ylim(-0.6, 1.0) +
  labs(title = "GO Vascular Wound Healing gene set - Car4+ ECs") +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank())
ggsave("Figures_Flu_pure_RJB/12_Car4_Endothelial_Module_VascHeal_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = c("Vwf+ endothelial cells")),
        features = c("Geneset_Vascular_healing1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(1.3, 0.8, 1.3)) +
  ylim(-0.6, 1.4) +
  theme(legend.position = "none") +
  labs(title = "GO Vascular Wound Healing geneset")
```

```{r Car4 ECs - DEGs}
# Flu_pure@active.ident <- Flu_pure$Timepoint_Age
# Flu_pure_DEGs_Car4Endo_TimeAge <- FindAllMarkers(subset(Flu_pure, Named_clusters == c("Car4+ endothelial cells")), only.pos = T)
# Flu_pure@active.ident <- Flu_pure$Named_clusters
# Flu_pure_DEGs_Car4Endo_TimeAge %>% group_by(cluster) %>% top_n(10, wt = avg_log2FC)

DotPlot(subset(Flu_pure, idents = c("Endothelial cells")),
        features = c("Cbr2", "Sdpr", "Cd97", "Cxcl10", "Egr1", "Icam1", "Irf1", "Atf3", "Ahr", "Clec2d", "Sox18", "Adgre5", "Klf2", "Foxf1", "Id1", "Vwf"),
        group.by = "Timepoint_Age") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r Car4 endothelial - VEGF module}
Geneset_VEGF_signaling <- read_tsv("Gene_sets/GOBP_VASCULAR_ENDOTHELIAL_GROWTH_FACTOR_SIGNALING_PATHWAY.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_VEGF_signaling <- paste0(
 substring(Geneset_VEGF_signaling, 1, 1),
 tolower(substring(Geneset_VEGF_signaling, 2))) # Change capitalization
Geneset_VEGF_signaling <- Geneset_VEGF_signaling[which(Geneset_VEGF_signaling %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_VEGF_signaling <- list(Geneset_VEGF_signaling)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_VEGF_signaling, name = "Geneset_VEGF_signaling")

VlnPlot(subset(Flu_pure, idents = c("Car4+ endothelial cells")),
        features = c("Geneset_VEGF_signaling1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.6, 0.6, 0.6)) +
  ylim(-0.2, 0.7) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank()) +
  labs(title = expression(bold("GO VEGF Signaling gene set - Car4" ^ "+" ~ "ECs")))
ggsave("Figures_Flu_pure_RJB/12_Car4_Endothelial_Module_VEGF_Vln.pdf")
```

```{r Epithelial - Wound response}
Geneset_Wound_response <- read_tsv("Gene_sets/GOBP_RESPONSE_TO_WOUNDING.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_Wound_response <- paste0(
 substring(Geneset_Wound_response, 1, 1),
 tolower(substring(Geneset_Wound_response, 2))) # Change capitalization
Geneset_Wound_response <- Geneset_Wound_response[which(Geneset_Wound_response %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_Wound_response <- list(Geneset_Wound_response)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_Wound_response, name = "Geneset_Wound_response")


DotPlot(Flu_pure, features = c("Geneset_Wound_response1"))
VlnPlot(Flu_pure, features = c("Geneset_Wound_response1"), pt.size = 0,
        # idents = c("Matrix fibroblasts", "Myofibroblasts", "Mesothelial cells", "AT1 cells", "AT2 cells", "Goblet cells", "Ciliated cells", "Club cells")) +
        idents = c("Matrix fibroblasts", "Myofibroblasts", "Mesothelial cells"),
        split.by = "Timepoint_Age") +
  NoLegend()


VlnPlot(subset(Flu_pure, idents = c("Matrix fibroblasts")),
        features = c("Geneset_Wound_response1"),
        group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.15, 0.225, 0.225)) +
  ylim(-0.01, 0.25) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5),
        axis.title = element_blank()) +
  labs(title = "GO Response to Wounding geneset - Matrix fibroblasts")
ggsave("Figures_Flu_pure_RJB/13_MatrixFibro_Module_WoundResponse_Vln.pdf")

# Differences not significant for myofibroblasts, mesothelial cells, or AT1/2/goblet/club/ciliated cells
```

```{r Gdf15}
# 10.1016/j.cell.2019.07.033

DotPlot(Flu_pure, features = c("Gdf15")) +
  theme(axis.title = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) +
  coord_flip()
ggsave("Figures_Flu_pure_RJB/14_Gdf15_DotPlot.pdf", width = 14, height = 5)
# Mostly expressed by myeloid and certain vascular cells (Car4+ ECs, pericytes)

VlnPlot(Flu_pure,
        features = c("Gdf15"), group.by = "Timepoint_Age", pt.size = 0.1,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(5, 5, 5)) +
  ylim(-0.01, 5.5)
ggsave("Figures_Flu_pure_RJB/14_Gdf15_TimeAge_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = "Alveolar macrophages"),
        features = c("Gdf15"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = c("Gdf15 - Alveolar macrophages")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.5, 4.5, 4.5)) +
  ylim(-0.01, 5)
ggsave("Figures_Flu_pure_RJB/14_Gdf15_AM_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = "Interstitial macrophages"),
        features = c("Gdf15"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = c("Gdf15 - Interstitial macrophages")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.5, 4.5, 4.5)) +
  ylim(-0.01, 5)
ggsave("Figures_Flu_pure_RJB/14_Gdf15_IntMac_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = "Car4+ endothelial cells"),
        features = c("Gdf15"), group.by = "Timepoint_Age", pt.size = 0.1,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = c("Gdf15 - Car4+ ECs")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.5, 4.5, 4.5)) +
  ylim(-0.01, 5)
ggsave("Figures_Flu_pure_RJB/14_Gdf15_Car4EC_Vln.pdf")

VlnPlot(subset(Flu_pure, idents = "Pericytes"),
        features = c("Gdf15"), group.by = "Timepoint_Age", pt.size = 0.1,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = c("Gdf15 - Pericytes")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.5, 4.5, 4.5)) +
  ylim(-0.01, 5)
```

```{r Type I IFNs}
Geneset_IFN_I <- read_tsv("Gene_sets/WP_INTERFERON_TYPE_I_SIGNALING_PATHWAYS.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_IFN_I <- paste0(
 substring(Geneset_IFN_I, 1, 1),
 tolower(substring(Geneset_IFN_I, 2))) # Change capitalization
Geneset_IFN_I <- Geneset_IFN_I[which(Geneset_IFN_I %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_IFN_I <- list(Geneset_IFN_I)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_IFN_I, name = "Geneset_IFN_I")

DotPlot(Flu_pure, features = c("Ifna1", "Ifnb1"))
# Ifna1 not detected, Ifnb1 is very low (< 1% of cells express it); Mostly in interstitial macrophages and monocytes

VlnPlot(subset(Flu_pure, idents = "Interstitial macrophages"),
        features = c("Ifnb1"), group.by = "Timepoint_Age", pt.size = 0.1,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = c("Ifnb1 - Interstitial macrophages")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.5, 4.5, 4.5)) +
  ylim(-0.01, 5)

VlnPlot(subset(Flu_pure, idents = "Monocytes"),
        features = c("Ifnb1"), group.by = "Timepoint_Age", pt.size = 0.1,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = c("Ifnb1 - Monocytes")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(4.5, 4.5, 4.5)) +
  ylim(-0.01, 5)


# IFN I module score
VlnPlot(subset(Flu_pure),
        features = c("Geneset_IFN_I1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = c("Type I IFN Signaling geneset")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.6, 0.6, 0.6)) +
  ylim(-0.3, 0.7)
```

```{r IFNg}
Geneset_IFNg <- read_tsv("Gene_sets/HALLMARK_INTERFERON_GAMMA_RESPONSE.v7.5.1.tsv") %>% 
  filter(STANDARD_NAME == "MAPPED_SYMBOLS") %>% 
  pull(2) %>% 
  strsplit(split = ",") %>% 
  unlist()
Geneset_IFNg <- paste0(
 substring(Geneset_IFNg, 1, 1),
 tolower(substring(Geneset_IFNg, 2))) # Change capitalization
Geneset_IFNg <- Geneset_IFNg[which(Geneset_IFNg %in% rownames(Flu_pure))] # Restrict to genes present in our data
Geneset_IFNg <- list(Geneset_IFNg)
Flu_pure <- AddModuleScore(Flu_pure, features = Geneset_IFNg, name = "Geneset_IFNg")


VlnPlot(subset(Flu_pure),
        features = c("Geneset_IFNg1"), group.by = "Timepoint_Age", pt.size = 0,
        cols = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  labs(title = c("IFN Gamma Signaling geneset")) +
  theme(axis.title = element_blank(),
        legend.position = "none",
        axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = ..y.., ymin = ..y..),
               width = 1, size = 1, linetype = "solid") +
  stat_compare_means(
    comparisons = list(
      c("Day 0 Young", "Day 0 Aged"),
      c("Day 3 Young", "Day 3 Aged"),
      c("Day 9 Young", "Day 9 Aged")),
    label.y = c(0.7, 0.7, 0.7)) +
  ylim(-0.2, 0.8)

DotPlot(Flu_pure, features = c("Geneset_IFNg1", "Ifng", "Ifngr1", "Ifngr2"))
```

```{r Macrophages}
DotPlot(Flu_pure, features = c("Ffar4"))
VlnPlot(subset(Flu_pure, idents = "Alveolar macrophages"), features = c("Ffar4"), group.by = "Timepoint_Age", pt.size = 0.1)
```

```{r DEGs from bulk RNA-seq}
# Cxcr6 elevated in aged D9 by bulk RNA-seq
DotPlot(Flu_pure, features = c("Cxcr6"))
# Only in gamma delta T cells
VlnPlot(subset(Flu_pure, idents = "Gamma delta T cells"), features = c("Cxcr6"), group.by = "Timepoint_Age", pt.size = 0)

# Serpins (esp Serpina1a/b/c/d aka A1AT) enriched in aged
DotPlot(Flu_pure, features = c("Serpina1a", "Serpina1b", "Serpina1c", "Serpina1d"))
DotPlot(Flu_pure, features = c("Serpina1a", "Serpina1b", "Serpina1c", "Serpina1d"), group.by = "Timepoint_Age")
```


```{r Cluster relative frequencies}
# Relative frequency between ages at each time point?
(Flu_pure@meta.data %>% filter(Named_clusters == "Suppressive neutrophils") %>% pull(Timepoint_Age) %>% table() %>% as.numeric()) / 
  (Flu_pure@meta.data %>% pull(Timepoint_Age) %>% table() %>% as.numeric()) * 100

Flu_pure@meta.data %>% filter(Named_clusters == "Suppressive neutrophils") %>% dplyr::select(Timepoint, Age) %>% table() %>% as.data.frame()
chisq.test(x = c(0, 7))
chisq.test(x = c(45, 67))
chisq.test(x = c(46, 117))

Flu_pure@meta.data %>%
  filter(Named_clusters == "Suppressive neutrophils" & Timepoint != "Day 0") %>%
  dplyr::select(Timepoint, Timepoint_Age) %>%
  table() %>% 
  prop.table(margin = 1) %>% 
  as.data.frame() %>%
  filter(!Timepoint_Age %in% c("Day 0 Young", "Day 0 Aged")) %>% 
  ggplot(aes(x = Timepoint, y = Freq, fill = Timepoint_Age)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Suppressive neutrophil frequency") +
  scale_fill_manual(name = "", values = c("lightskyblue3", "royalblue3", "orange", "orangered3")) +
  # theme(legend.position = "none") +
  geom_hline(yintercept = 0.5, color = "firebrick1") +
  annotate(
    geom = "text",
    x = "Day 3",
    y = 0.33,
    label = expression(italic("p") ~ "= 0.04"),
    size = 4,
    color = "white") +
  annotate(
    geom = "text",
    x = "Day 9",
    y = 0.33,
    label = expression(italic("p") ~ "= 2.7 x 10" ^ "-8"),
    size = 4,
    color = "white")
ggsave("Figures_Flu_pure_RJB/04_Suppressive_neutrophil_freq.pdf", width = 5, height = 4)



(Flu_pure@meta.data %>% filter(Named_clusters == "Car4+ endothelial cells") %>% pull(Timepoint_Age) %>% table() %>% as.numeric()) / 
  (Flu_pure@meta.data %>% pull(Timepoint_Age) %>% table() %>% as.numeric()) * 100

Flu_pure@meta.data %>% filter(Named_clusters == "Car4+ endothelial cells") %>% pull(Timepoint_Age) %>% table() %>% as.numeric()
chisq.test(x = c(85, 110))
chisq.test(x = c(255, 417))
chisq.test(x = c(361, 288))

Flu_pure@meta.data %>%
  filter(Named_clusters == "Car4+ endothelial cells") %>%
  dplyr::select(Timepoint, Timepoint_Age) %>%
  table() %>% 
  prop.table(margin = 1) %>% 
  as.data.frame() %>%
  ggplot(aes(x = Timepoint, y = Freq, fill = Timepoint_Age)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Car4+ endothelial cell frequency") +
  scale_fill_manual(name = "", values = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  geom_hline(yintercept = 0.5, color = "firebrick1") +
  annotate(
    geom = "text",
    x = "Day 0",
    y = 0.33,
    label = expression("ns"),
    size = 4,
    color = "white") +
  annotate(
    geom = "text",
    x = "Day 3",
    y = 0.33,
    label = expression(italic("p") ~ "= 4.1 x 10" ^ "-10"),
    size = 4,
    color = "white") +
  annotate(
    geom = "text",
    x = "Day 9",
    y = 0.33,
    label = expression(italic("p") ~ "= 0.004"),
    size = 4,
    color = "white")
ggsave("Figures_Flu_pure_RJB/04_Car4EC_freq.pdf", width = 6, height = 4)





Flu_pure@meta.data %>% filter(Named_clusters == "Naive B cells") %>% dplyr::select(Timepoint, Age) %>% table() %>% as.data.frame()
chisq.test(x = c(258, 443))
chisq.test(x = c(1412, 1807))
chisq.test(x = c(443, 537))

Flu_pure@meta.data %>%
  filter(Named_clusters == "Naive B cells") %>%
  dplyr::select(Timepoint, Timepoint_Age) %>%
  table() %>% 
  prop.table(margin = 1) %>% 
  as.data.frame() %>%
  ggplot(aes(x = Timepoint, y = Freq, fill = Timepoint_Age)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Naive B cell frequency") +
  scale_fill_manual(name = "", values = c("gray70", "gray20", "lightskyblue3", "royalblue3", "orange", "orangered3")) +
  geom_hline(yintercept = 0.5, color = "firebrick1") +
  annotate(
    geom = "text",
    x = "Day 0",
    y = 0.33,
    label = expression(italic("p") ~ "= 2.8 x 10" ^ "-12"),
    size = 4,
    color = "white") +
  annotate(
    geom = "text",
    x = "Day 3",
    y = 0.33,
    label = expression(italic("p") ~ "= 3.4 x 10" ^ "-12"),
    size = 4,
    color = "white") +
  annotate(
    geom = "text",
    x = "Day 9",
    y = 0.33,
    label = expression(italic("p") ~ "= 0.003"),
    size = 4,
    color = "white")
ggsave("Figures_Flu_pure_RJB/04_NaiveBcell_freq.pdf", width = 6, height = 4)
```

```{r F13a1}
DotPlot(Flu_pure, features = c("F13a1", "Plat", "Serpine1"))
```








```{r T cells - subset}
Flu_pure_Tcells <- subset(Flu_pure, idents = c("CD4 T cells", "CD8 T cells"))

Flu_more_pure_Tcells <- subset(Flu_pure_Tcells, idents = c("4", "5"), invert = T)

table(Flu_pure_Tcells@active.ident)
table(Flu_pure_Tcells$Timepoint)
table(Flu_pure_Tcells$Timepoint_Age)
```

```{r T cells - UMAP}
DefaultAssay(Flu_pure_Tcells) <- "integrated"

Flu_pure_Tcells <- ScaleData(Flu_pure_Tcells, features = rownames(Flu_pure_Tcells))
Flu_pure_Tcells <- FindVariableFeatures(Flu_pure_Tcells)

Flu_pure_Tcells <- RunPCA(Flu_pure_Tcells, npcs = 50, verbose = F)
Flu_pure_Tcells <- RunUMAP(Flu_pure_Tcells, reduction = "pca", dims = 1:50)
Flu_pure_Tcells <- FindNeighbors(Flu_pure_Tcells, reduction = "pca", dims = 1:50)
Flu_pure_Tcells <- FindClusters(Flu_pure_Tcells, resolution = 0.2)

DimPlot(Flu_pure_Tcells)
DimPlot(Flu_pure_Tcells, split.by = "Timepoint_Age", ncol = 2)
DimPlot(Flu_pure_Tcells, split.by = "Age")
```



```{r}

Flu_pure_Tcells <- subset(Flu_pure_Tcells, idents = c(4, 5), invert = T)


DefaultAssay(Flu_pure_Tcells) <- "integrated"

Flu_pure_Tcells <- ScaleData(Flu_pure_Tcells, features = rownames(Flu_pure_Tcells))
Flu_pure_Tcells <- FindVariableFeatures(Flu_pure_Tcells)

Flu_pure_Tcells <- RunPCA(Flu_pure_Tcells, npcs = 50, verbose = F)
Flu_pure_Tcells <- RunUMAP(Flu_pure_Tcells, reduction = "pca", dims = 1:50)
Flu_pure_Tcells <- FindNeighbors(Flu_pure_Tcells, reduction = "pca", dims = 1:50)
Flu_pure_Tcells <- FindClusters(Flu_pure_Tcells, resolution = 0.2)


Flu_pure_Tcells <- RenameIdents(Flu_pure_Tcells,
                        "0" = "Memory CD8",
                        "1" = "Effector CD8",
                        "2" = "Memory CD4",
                        "3" = "Effector CD4",
                        "4" = "Memory CD8"
                        )

DimPlot(Flu_pure_Tcells)
ggsave("Figures_Flu_pure_RJB/Tcell_Only/01_Umap.pdf", width = 6, height = 4)

DimPlot(Flu_pure_Tcells, split.by = "Timepoint_Age", ncol = 2)
DimPlot(Flu_pure_Tcells, split.by = "Age")

saveRDS(Flu_pure_Tcells, "RDS_files.nosync/Flu_pure_Tcells_RJB.RDS")
Flu_pure_Tcells <- readRDS("RDS_files.nosync/Flu_pure_Tcells_RJB.RDS")
Flu_pure_Tcells <- RenameIdents(Flu_pure_Tcells,
                        "Memory CD8" = "CD8 MPC",
                        "Effector CD8" = "CD8 Effector",
                        "Memory CD4" = "CD4 MPC",
                        "Effector CD4" = "CD4 Effector"
                        )

```








```{r T cells - markers}
DefaultAssay(Flu_pure_Tcells) <- "RNA"

DotPlot(Flu_pure_Tcells, features = c("Cd3e", "Cd4", "Cd8a", "Cd44", "Sell", "Il7r", "Klrk1", "Gzmb", "Gata3", "Foxp3",  "Ccr7", "Klf3", "Bhlhe40", "Id2")) +RotatedAxis()
ggsave("Figures_Flu_pure_RJB/Tcell_Only/02_DotPlot.pdf", width = 6, height = 4)

DotPlot(Flu_pure_Tcells, features = c("Cd3e", "Cd4", "Cd8a", "Cd44", "Cxcr5", "Cxcr3", "Tbx21", "Bcl6", "Slamf1", "Il4", "Il5", "Il13"))


FeaturePlot(Flu_pure_Tcells, features = c('Cd8a'), order = T)
FeaturePlot(Flu_pure_Tcells, features = c('Cd4'), order = T)
FeaturePlot(Flu_pure_Tcells, features = c('Foxp3'))
FeaturePlot(Flu_pure_Tcells, features = c('Cxcr5'))
FeaturePlot(Flu_pure_Tcells, features = c('Cxcr3'))
FeaturePlot(Flu_pure_Tcells, features = c('Cxcr3'))
FeaturePlot(Flu_pure_Tcells, features = c('Ccr5'))
FeaturePlot(Flu_pure_Tcells, features = c('Cd44'), order = T)

DotPlot(Flu_pure_Tcells, features = c("Tbx21", "Cd4", "Cd8a"))

Flu_pure_Tcells_markers <- FindAllMarkers(Flu_pure_Tcells, only.pos = T)
Flu_pure_Tcells_markers %>% group_by(cluster) %>% top_n(20, wt = avg_log2FC)

# Naive T cells (1 and 2) suggest we may have harvested some lymph nodes
```



```{r}
DefaultAssay(Flu_pure_Tcells) <- "RNA"

VlnPlot(subset(Flu_pure_Tcells, subset = Timepoint == "Day 9"), features = c("Klf3"), pt.size = 0, split.by = "Timepoint_Age", split.plot = T) # Test labeling

VlnPlot(subset(Flu_pure_Tcells, subset = Timepoint == "Day 9"), features = c("Klf3", "Ccr7", "Id2", "Bhlhe40"), pt.size = 0, split.by = "Timepoint_Age", split.plot = T) # CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)
VlnPlot(subset(subset(Flu_pure_Tcells, ident = c("CD4 MPC")), subset = Timepoint == "Day 9"), features = c("Klf3"), pt.size = 0, split.by = "Timepoint_Age", split.plot = F) + ggplot2::scale_fill_manual(values = c("orange", "orangered3")) # CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)
ggsave("Figures_Flu_pure_RJB/Tcell_Only/04_Klf3.pdf", width = 6, height = 4)

VlnPlot(subset(subset(Flu_pure_Tcells, ident = c("CD4 MPC")), subset = Timepoint == "Day 9"), features = c("Ccr7"), pt.size = 0, split.by = "Timepoint_Age", split.plot = F) + ggplot2::scale_fill_manual(values = c("orange", "orangered3")) # CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)
ggsave("Figures_Flu_pure_RJB/Tcell_Only/05_Ccr7.pdf", width = 6, height = 4)




VlnPlot(subset(subset(Flu_pure_Tcells, ident = c("CD4 Effector")), subset = Timepoint == "Day 9"), features = c( "Id2"), pt.size = 0, split.by = "Timepoint_Age", split.plot = F) + ggplot2::scale_fill_manual(values = c("orange", "orangered3"))# CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)
ggsave("Figures_Flu_pure_RJB/Tcell_Only/06_Id2.pdf", width = 6, height = 4)


VlnPlot(subset(subset(Flu_pure_Tcells, ident = c("CD4 Effector")), subset = Timepoint == "Day 9"), features = c("Bhlhe40"), pt.size = 0, split.by = "Timepoint_Age", split.plot = F) + ggplot2::scale_fill_manual(values = c("orange", "orangered3"))# CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)
ggsave("Figures_Flu_pure_RJB/Tcell_Only/06_Bhlhe40.pdf", width = 6, height = 4)


VlnPlot(subset(Flu_pure_Tcells, subset = Timepoint == "Day 9"), features = c("Bach2", "Lef1", "Tcf7"), pt.size = 0, split.by = "Age", split.plot = T) # CD8 memory
VlnPlot(subset(subset(Flu_pure_Tcells, ident = c("CD8 MPC")), subset = Timepoint == "Day 9"), features = c("Bach2", "Lef1", "Tcf7"), pt.size = 0, split.by = "Age", split.plot = F, cols = c("orange", "orangered3")) # CD8 memory
ggsave("Figures_Flu_pure_RJB/Tcell_Only/07_Bach2_Tcf7_Lef1.pdf", width = 6, height = 4)


VlnPlot(subset(subset(Flu_pure_Tcells, ident = c("CD8 MPC", "CD8 Effector")), subset = Timepoint == "Day 9"), features = c("Ccr5"), pt.size = 0, split.by = "Age", split.plot = F) & ggplot2::scale_fill_manual(values = c("orange", "orangered3"))
ggsave("Figures_Flu_pure_RJB/Tcell_Only/08_Ccr7.pdf", width = 6, height = 4)

table(Flu_more_pure_Tcells@active.ident, Flu_more_pure_Tcells$Timepoint)
```



```{r}
table(Flu_pure_Tcells@active.ident, Flu_pure_Tcells$Timepoint_Age) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  xlab("Age") # +
#    geom_text(
#    aes(
#      y = Freq,
#      label = ifelse(
#        Freq > 0.025,
#        paste0(Cluster),
#        "")),
#    position = position_stack(vjust = 0.5),
#    size = 2)
ggsave("Figures_Flu_pure_RJB/Tcell_Only/03_StackedBoxPlot.pdf", width = 6, height = 4)

# Find number of 0 in all aged / young cells cells

print(packageVersion("Seurat"))

table(Flu_pure_Tcells@active.ident, Flu_pure_Tcells$Timepoint)
table(Flu_pure_Tcells$Timepoint, Flu_pure_Tcells@active.ident)

```






```{r Split into CD8}
Flu_pure_Tcells <- readRDS('RDS_files.nosync/Flu_pure_Tcells_RJB.RDS')
Flu_CD8 <- subset(Flu_pure_Tcells, idents = c("CD8 MPC", "CD8 Effector"))

DefaultAssay(Flu_CD8) <- "integrated"

Flu_CD8 <- ScaleData(Flu_CD8, features = rownames(Flu_CD8))
Flu_CD8 <- FindVariableFeatures(Flu_CD8)

Flu_CD8 <- RunPCA(Flu_CD8, npcs = 50, verbose = F)
Flu_CD8 <- RunUMAP(Flu_CD8, reduction = "pca", dims = 1:50)
Flu_CD8 <- FindNeighbors(Flu_CD8, reduction = "pca", dims = 1:50)
Flu_CD8 <- FindClusters(Flu_CD8, resolution = 0.2)

Flu_CD8_3_9 <- subset(subset(Flu_CD8, subset = Timepoint ==  "Day 0", invert = T),  ident = c(3, 4), invert = T)
Flu_CD8_3_9 <- RenameIdents(Flu_CD8_3_9,
                  "0" = "CD8 Effector",
                  "1" = 'CD8 Naive',
                  "2" = "CD8 MPC"
)
DimPlot(Flu_CD8_3_9)
ggsave("Figures_Flu_pure_RJB/CD8_Only/01_UMAP.pdf", width = 6, height = 4)




# Create Cluster identify column
Flu_CD8_3_9@meta.data$ClusterName <- Flu_CD8_3_9@active.ident


# Create Cluster and identity column
Flu_CD8_3_9@meta.data$Cluster_Age <- ""
for(i in 1:length(Flu_CD8_3_9@meta.data$Cluster_Age)){
  Flu_CD8_3_9@meta.data$Cluster_Age[i] = paste(Flu_CD8_3_9@meta.data$Age[i], Flu_CD8_3_9@meta.data$ClusterName[i])
}


Flu_CD8_3_9$Cluster_Age<- factor(x = Flu_CD8_3_9$Cluster_Age, levels = c("Young CD8 Naive", "Aged CD8 Naive", "Young CD8 MPC", "Aged CD8 MPC", "Young CD8 Effector", "Aged CD8 Effector"))


```

```{r CD8}


markers <- FindAllMarkers(Flu_CD8, only.pos = T)

DefaultAssay(Flu_CD8_3_9) <- "RNA"
DotPlot(Flu_CD8_3_9, features = c("Cd8a", "Cd44", "Sell", "Il7r", "Lef1", "Tcf7", "Gzma", "Gzmb", "Zeb2", "Klrg1", "Cx3cr1")) + RotatedAxis()
ggsave("Figures_Flu_pure_RJB/CD8_Only/02_Dotplot.pdf", width = 6, height = 4)


FeaturePlot(Flu_CD8_3_9, features = c("Cd44"), order = T)
Flu_CD8_3_9$Timepoint_Age <- factor(Flu_CD8_3_9$Timepoint_Age)

table(Flu_CD8_3_9@active.ident, Flu_CD8_3_9$Timepoint_Age) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  xlab("Age")
ggsave("Figures_Flu_pure_RJB/CD8_Only/03_BarGraph.pdf", width = 6, height = 4)

```

```{r VlnPlots}

VlnPlot(subset(Flu_CD8_3_9, subset = Timepoint == "Day 9"), features = c("Bach2", "Lef1", "Tcf7"), pt.size = 0, split.by = "Age", split.plot = T) # CD8 memory
VlnPlot(subset(subset(Flu_CD8_3_9, ident = c("CD8 MPC")), subset = Timepoint == "Day 9"), features = c("Tcf7"), pt.size = 0, split.by = "Age", split.plot = F, cols = c("orange", "orangered3")) # CD8 memory
ggsave("Figures_Flu_pure_RJB/CD8_Only/04_Tcf7.pdf", width = 6, height = 4)


VlnPlot(subset(subset(Flu_CD8_3_9, ident = c("CD8 MPC", "CD8 Effector")), subset = Timepoint == "Day 9"), features = c("Ccr5"), pt.size = 0, split.by = "Age", split.plot = F) & ggplot2::scale_fill_manual(values = c("orange", "orangered3"))
ggsave("Figures_Flu_pure_RJB/CD8_Only/05_Ccr7.pdf", width = 6, height = 4)

table(Flu_more_pure_Tcells@active.ident, Flu_more_pure_Tcells$Timepoint)
```




```{r Complex Heatmap CD8}
set.seed(123)
# Set the identity of your cells to the desired column
Idents(object = Flu_CD8_3_9) <- Flu_CD8_3_9@meta.data$'Cluster_Age'


#install_github("jokergoo/ComplexHeatmap")
library(ComplexHeatmap)

DefaultAssay(Flu_CD8_3_9) <- 'RNA'
RNA_markers <- FindAllMarkers(subset(Flu_CD8_3_9, subset = Timepoint == "Day 9"), only.pos = TRUE, min.pct = .1, logfc.threshold = .1)
RNA_markers <- RNA_markers[RNA_markers$p_val_adj < .05,]
RNA_markers <- RNA_markers[!grepl('Rpl', RNA_markers$gene),]
RNA_markers <- RNA_markers[!grepl('Rps', RNA_markers$gene),]



averaged_RNA <- AverageExpression(Flu_CD8_3_9, features = RNA_markers$gene, slot = "data", assays = "RNA")
averaged_RNA <- as.data.frame(averaged_RNA$RNA)
averaged_RNA <- t(scale(t(averaged_RNA)))
averaged_RNA <- as.data.frame(averaged_RNA)

# We need a vector of all genes in the heatmap
allnames = rownames(averaged_RNA)
# This is a vector of all genes we would like to have labeled
genes_to_label = c( "Tox", "Eomes", "Nr4a2", "Pdcd1",'Cd244a', 'Il10ra',"Cx3cr1", "Tbx21", "Zeb2", 'Klrg1', 'Pfn1','Ifng', "Sell", "Tcf7", "Ccr7", "Bik",'Bcl2l10', 'Bcl2l11', 'Bcl2l2', 'Casp2','Casp3','Casp4','Fas', "Fli1", 'Cd101', "Gzmb", "Gzma", "Bach2", "Lef1", "Tcf7", "Ccr5", "Il7r", 'Xcl1', 'E4f1', "Fau", 'Cdk2ap2', 'Klf4')


# Find index that match genes in the list
index_w_genes <- c()
genes_not_found <- c()
for(i in 1:length(genes_to_label)){
  for(j in 1:length(allnames)){
    if(allnames[j] == genes_to_label[i]){
      index_w_genes <- append(index_w_genes, j)
      break
    } else {
      if(j == length(allnames)){
        genes_not_found <- append(genes_not_found, genes_to_label[i])
      }
    }
  }
}

# Remove any gene not found in the heatmap
genes_to_label <- setdiff(genes_to_label, genes_not_found)

# make heatmap
ra <- rowAnnotation(foo = anno_mark(index_w_genes, labels =  genes_to_label, labels_gp = gpar(fontsize = 14), padding = unit(.1, "mm")))
pdf(file = "Figures_Flu_pure_RJB/CD8_Only/06_complex_heatmap_RNA.pdf")
Heatmap(averaged_RNA, cluster_columns = F, cluster_rows = F, right_annotation = ra, show_row_names = F, column_names_rot = 65, column_names_centered = F, column_names_gp = grid::gpar(fontsize = 15), name = "z-score", km = 5, row_km = 5, column_labels = c("Young CD8 Naive", "Aged CD8 Naive", "Young CD8 MPC", "Aged CD8 MPC", "Young CD8 Effector", "Aged CD8 Effector")) # column_labels = c("Tpro1", "Tpro2", "Teff", "Texh")
dev.off()


ht = draw(h)
ht = row_order(ht) # a list of two 
rownames(averaged_RNA)[ ht[[3]] ] # the gene ids in the first cluster

```

```{r Complex Heatmap CD8 day 3 and 9}
set.seed(123)
# Set the identity of your cells to the desired column
Idents(object = Flu_CD8_3_9) <- Flu_CD8_3_9@meta.data$'Cluster_Age'


#install_github("jokergoo/ComplexHeatmap")
library(ComplexHeatmap)

DefaultAssay(Flu_CD8_3_9) <- 'RNA'
RNA_markers <- FindAllMarkers(Flu_CD8_3_9, only.pos = TRUE, min.pct = .1, logfc.threshold = .1)
RNA_markers <- RNA_markers[RNA_markers$p_val_adj < .05,]
RNA_markers <- RNA_markers[!grepl('Rpl', RNA_markers$gene),]
RNA_markers <- RNA_markers[!grepl('Rps', RNA_markers$gene),]



averaged_RNA <- AverageExpression(Flu_CD8_3_9, features = RNA_markers$gene, slot = "data", assays = "RNA")
averaged_RNA <- as.data.frame(averaged_RNA$RNA)
averaged_RNA <- t(scale(t(averaged_RNA)))
averaged_RNA <- as.data.frame(averaged_RNA)

# We need a vector of all genes in the heatmap
allnames = rownames(averaged_RNA)
# This is a vector of all genes we would like to have labeled
genes_to_label = c( "Tox", "Eomes", "Nr4a2", "Pdcd1",'Cd244a', 'Il10ra',"Cx3cr1", "Tbx21", "Zeb2", 'Klrg1', 'Pfn1','Ifng', "Sell", "Tcf7", "Ccr7", "Bik",'Bcl2l10', 'Bcl2l11', 'Bcl2l2', 'Casp2','Casp3','Casp4','Fas', "Fli1", 'Cd101', "Gzmb", "Gzma", "Bach2", "Lef1", "Tcf7", "Ccr5", "Il7r", 'Xcl1', 'E4f1', "Fau", 'Cdk2ap2', 'Klf4', 'Il21r', 'Klf2', 'Jun')


# Find index that match genes in the list
index_w_genes <- c()
genes_not_found <- c()
for(i in 1:length(genes_to_label)){
  for(j in 1:length(allnames)){
    if(allnames[j] == genes_to_label[i]){
      index_w_genes <- append(index_w_genes, j)
      break
    } else {
      if(j == length(allnames)){
        genes_not_found <- append(genes_not_found, genes_to_label[i])
      }
    }
  }
}

# Remove any gene not found in the heatmap
genes_to_label <- setdiff(genes_to_label, genes_not_found)

# make heatmap
ra <- rowAnnotation(foo = anno_mark(index_w_genes, labels =  genes_to_label, labels_gp = gpar(fontsize = 14), padding = unit(.1, "mm")))
pdf(file = "Figures_Flu_pure_RJB/CD8_Only/06_complex_heatmap_RNA_3_9.pdf")
Heatmap(averaged_RNA, cluster_columns = F, cluster_rows = F, right_annotation = ra, show_row_names = F, column_names_rot = 65, column_names_centered = F, column_names_gp = grid::gpar(fontsize = 15), name = "z-score", km = 5, row_km = 5,column_labels = c("Young CD8 Naive", "Aged CD8 Naive", "Young CD8 MPC", "Aged CD8 MPC", "Young CD8 Effector", "Aged CD8 Effector")) # column_labels = c("Tpro1", "Tpro2", "Teff", "Texh")
dev.off()


ht = draw(h)
ht = row_order(ht) # a list of two 
for(i in 1:length(ht)){
  print(i)
  print(rownames(averaged_RNA)[ ht[[i]] ]) # the gene ids in the first cluster

}


```





```{r Split into CD4}

Flu_CD4 <- subset(Flu_pure_Tcells, idents = c("CD4 MPC", "CD4 Effector"))


DefaultAssay(Flu_CD4) <- "integrated"

Flu_CD4 <- ScaleData(Flu_CD4, features = rownames(Flu_CD4))
Flu_CD4 <- FindVariableFeatures(Flu_CD4)

Flu_CD4 <- RunPCA(Flu_CD4, npcs = 50, verbose = F)
Flu_CD4 <- RunUMAP(Flu_CD4, reduction = "pca", dims = 1:50)
Flu_CD4 <- FindNeighbors(Flu_CD4, reduction = "pca", dims = 1:50)
Flu_CD4 <- FindClusters(Flu_CD4, resolution = 0.2)
DimPlot(Flu_CD4)


Flu_CD4_3_9 <- subset(subset(Flu_CD4, subset = Timepoint ==  "Day 0", invert = T),  ident = c(3, 4), invert = T)
Flu_CD4_3_9 <- RenameIdents(Flu_CD4_3_9,
                  "0" = "CD4 Effector",
                  "1" = 'CD4 Naive',
                  "2" = "CD4 MPC"
)
DimPlot(Flu_CD4_3_9)
ggsave("Figures_Flu_pure_RJB/CD4_Only/01_UMAP.pdf", width = 6, height = 4)


Flu_CD4_3_9$Timepoint_Age <- factor(Flu_CD4_3_9$Timepoint_Age)
table(Flu_CD4_3_9@active.ident, Flu_CD4_3_9$Timepoint_Age) %>%
  prop.table(margin = 2) %>%
  as.data.frame() %>% 
  dplyr::rename(Cluster = Var1, Sample = Var2) %>% 
  ggplot(aes(x = Sample, y = Freq, fill = Cluster)) +
  geom_bar(stat = "identity", position = "stack") +
  theme_minimal() +
  theme(panel.grid.major.x = element_blank()) +
  scale_y_continuous(labels = scales::percent_format()) +
  ylab("Frequency") +
  xlab("Age")
ggsave("Figures_Flu_pure_RJB/CD4_Only/02_BarGraph.pdf", width = 6, height = 4)






# Create Cluster identify column
Flu_CD4_3_9@meta.data$ClusterName <- Flu_CD4_3_9@active.ident


# Create Cluster and identity column
Flu_CD4_3_9@meta.data$Cluster_Age <- ""
for(i in 1:length(Flu_CD4_3_9@meta.data$Cluster_Age)){
  Flu_CD4_3_9@meta.data$Cluster_Age[i] = paste(Flu_CD4_3_9@meta.data$Age[i], Flu_CD4_3_9@meta.data$ClusterName[i])
}


Flu_CD4_3_9$Cluster_Age <- factor(x = Flu_CD4_3_9$Cluster_Age, levels = c("Young CD4 Naive", "Aged CD4 Naive", "Young CD4 MPC", "Aged CD4 MPC", "Young CD4 Effector", "Aged CD4 Effector"))

```


```{r}

markers <- FindAllMarkers(Flu_CD4, only.pos = T)
markers <- markers %>% group_by(cluster) %>% top_n(100, avg_log2FC)

DefaultAssay(Flu_CD4_3_9) <- "RNA"
DotPlot(Flu_CD4_3_9, features = c("Cd4", "Cd44", "Sell", "Il7r", "Lef1", "Tcf7")) + RotatedAxis()
ggsave("Figures_Flu_pure_RJB/CD4_Only/03_DotPlot.pdf", width = 6, height = 4)



FeaturePlot(Flu_CD4_3_9, features = c("Gzmb"), order = T)



```



```{r VlnPlots}

DefaultAssay(Flu_CD4_3_9) <- "RNA"



VlnPlot(subset(Flu_CD4_3_9, subset = Timepoint == "Day 9"), features = c("Klf3", "Ccr7", "Id2", "Bhlhe40"), pt.size = 0, split.by = "Timepoint_Age", split.plot = T) # CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)

VlnPlot(subset(subset(Flu_CD4_3_9, ident = c("MPC")), subset = Timepoint == "Day 9"), features = c("Klf3"), pt.size = 0, split.by = "Timepoint_Age", split.plot = F) + ggplot2::scale_fill_manual(values = c("orange", "orangered3")) # CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)

VlnPlot(subset(subset(Flu_CD4_3_9, ident = c("MPC")), subset = Timepoint == "Day 9"), features = c("Ccr7"), pt.size = 0, split.by = "Timepoint_Age", split.plot = F) + ggplot2::scale_fill_manual(values = c("orange", "orangered3")) # CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)




VlnPlot(subset(subset(Flu_CD4_3_9, ident = c("CD4 Effector")), subset = Timepoint == "Day 9"), features = c( "Id2"), pt.size = 0, split.by = "Timepoint_Age", split.plot = F) + ggplot2::scale_fill_manual(values = c("orange", "orangered3"))# CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)


VlnPlot(subset(subset(Flu_CD4_3_9, ident = c("CD4 Effector")), subset = Timepoint == "Day 9"), features = c("Bhlhe40"), pt.size = 0, split.by = "Timepoint_Age", split.plot = F) + ggplot2::scale_fill_manual(values = c("orange", "orangered3"))# CD4 (Klf3/Ccr7 == mem, Id2/Bhlhe40 == Th1)
ggsave("Figures_Flu_pure_RJB/CD4_Only/04_Bhlhe40.pdf", width = 6, height = 4)


```


```{r}
set.seed(123)
# Set the identity of your cells to the desired column
Idents(object = Flu_CD4_3_9) <- Flu_CD4_3_9@meta.data$'Cluster_Age'


#install_github("jokergoo/ComplexHeatmap")
library(ComplexHeatmap)

DefaultAssay(Flu_CD4_3_9) <- 'RNA'
RNA_markers <- FindAllMarkers(subset(Flu_CD4_3_9, subset = Timepoint == "Day 9"), only.pos = TRUE, min.pct = .1, logfc.threshold = .1)
RNA_markers <- RNA_markers[RNA_markers$p_val_adj < .05,]
RNA_markers <- RNA_markers[!grepl('Rpl', RNA_markers$gene),]
RNA_markers <- RNA_markers[!grepl('Rps', RNA_markers$gene),]



averaged_RNA <- AverageExpression(Flu_CD4_3_9, features = RNA_markers$gene, slot = "data", assays = "RNA")
averaged_RNA <- as.data.frame(averaged_RNA$RNA)
averaged_RNA <- t(scale(t(averaged_RNA)))
averaged_RNA <- as.data.frame(averaged_RNA)

# We need a vector of all genes in the heatmap
allnames = rownames(averaged_RNA)
# This is a vector of all genes we would like to have labeled
genes_to_label = c(  "Nr4a2", "Pdcd1", 'Il10ra','Ifng', "Sell", "Tcf7", "Ccr7", 'Bcl2', 'Casp2','Casp3','Casp4', "Bach2", "Lef1", "Tcf7", "Ccr5", "Il7r", 'Xcl1', 'E4f1', 'Cdk2ap2', "Tbx21", "Klf3", "Ccr7", "Id2", "Bhlhe40", "Il21", "Il4", "Il6", "Irf7", "Irf2", "Stat1", "Ifih1", 'Fosb', "Fos", "Zap70", "Junb", "Eomes", 'Cdkn1a', 'Il2ra', "S100a4", "S100a11", "Lgals3", "Tox", "")


# Find index that match genes in the list
index_w_genes <- c()
genes_not_found <- c()
for(i in 1:length(genes_to_label)){
  for(j in 1:length(allnames)){
    if(allnames[j] == genes_to_label[i]){
      index_w_genes <- append(index_w_genes, j)
      break
    } else {
      if(j == length(allnames)){
        genes_not_found <- append(genes_not_found, genes_to_label[i])
      }
    }
  }
}

# Remove any gene not found in the heatmap
genes_to_label <- setdiff(genes_to_label, genes_not_found)

# make heatmap
ra <- rowAnnotation(foo = anno_mark(index_w_genes, labels =  genes_to_label, labels_gp = gpar(fontsize = 14), padding = unit(.1, "mm")))
pdf(file = "Figures_Flu_pure_RJB/CD4_Only/06_complex_heatmap_RNA.pdf")
Heatmap(averaged_RNA, cluster_columns = F, cluster_rows = F, right_annotation = ra, show_row_names = F, column_names_rot = 65, column_names_centered = F, column_names_gp = grid::gpar(fontsize = 15), name = "z-score", km = 5, row_km = 5, column_labels = c("Young CD4 Naive", "Aged CD4 Naive", "Young CD4 MPC", "Aged CD4 MPC", "Young CD4 Effector", "Aged CD4 Effector")) # column_labels = c("Tpro1", "Tpro2", "Teff", "Texh")
dev.off()


ht = draw(h)
ht = row_order(ht) # a list of two 

for(i in 1:length(ht)){
  print(i)
  print(rownames(averaged_RNA)[ ht[[i]] ]) # the gene ids in the first cluster

}


```

```{r Complex Heatmap CD4 day 3 and 9}
set.seed(123)
# Set the identity of your cells to the desired column
Idents(object = Flu_CD4_3_9) <- Flu_CD4_3_9@meta.data$'Cluster_Age'


#install_github("jokergoo/ComplexHeatmap")
library(ComplexHeatmap)

DefaultAssay(Flu_CD4_3_9) <- 'RNA'
RNA_markers <- FindAllMarkers(Flu_CD4_3_9, only.pos = TRUE, min.pct = .1, logfc.threshold = .1)
RNA_markers <- RNA_markers[RNA_markers$p_val_adj < .05,]
RNA_markers <- RNA_markers[!grepl('Rpl', RNA_markers$gene),]
RNA_markers <- RNA_markers[!grepl('Rps', RNA_markers$gene),]



averaged_RNA <- AverageExpression(Flu_CD4_3_9, features = RNA_markers$gene, slot = "data", assays = "RNA")
averaged_RNA <- as.data.frame(averaged_RNA$RNA)
averaged_RNA <- t(scale(t(averaged_RNA)))
averaged_RNA <- as.data.frame(averaged_RNA)

# We need a vector of all genes in the heatmap
allnames = rownames(averaged_RNA)
# This is a vector of all genes we would like to have labeled
genes_to_label = c(  "Nr4a2", "Pdcd1", 'Il10ra','Ifng', "Sell", "Tcf7", "Ccr7", 'Bcl2', 'Casp2','Casp3','Casp4', "Bach2", "Lef1", "Tcf7", "Ccr5", "Il7r", 'Xcl1', 'E4f1', 'Cdk2ap2', "Tbx21", "Klf3", "Ccr7", "Id2", "Bhlhe40", "Il21", "Il4", "Il6", "Irf7", "Irf2", "Stat1", "Ifih1", 'Fosb', "Fos", "Zap70", "Junb", "Eomes", 'Cdkn1a', 'Il2ra', "S100a4", "S100a11", "Lgals3", "Tox", "Cd44")


# Find index that match genes in the list
index_w_genes <- c()
genes_not_found <- c()
for(i in 1:length(genes_to_label)){
  for(j in 1:length(allnames)){
    if(allnames[j] == genes_to_label[i]){
      index_w_genes <- append(index_w_genes, j)
      break
    } else {
      if(j == length(allnames)){
        genes_not_found <- append(genes_not_found, genes_to_label[i])
      }
    }
  }
}

# Remove any gene not found in the heatmap
genes_to_label <- setdiff(genes_to_label, genes_not_found)

# make heatmap
ra <- rowAnnotation(foo = anno_mark(index_w_genes, labels =  genes_to_label, labels_gp = gpar(fontsize = 14), padding = unit(.1, "mm")))
pdf(file = "Figures_Flu_pure_RJB/CD4_Only/06_complex_heatmap_RNA_3_9.pdf")
Heatmap(averaged_RNA, cluster_columns = F, cluster_rows = F, right_annotation = ra, show_row_names = F, column_names_rot = 65, column_names_centered = F, column_names_gp = grid::gpar(fontsize = 15), name = "z-score", km = 5, row_km = 5, column_labels = c("Young CD4 Naive", "Aged CD4 Naive", "Young CD4 MPC", "Aged CD4 MPC", "Young CD4 Effector", "Aged CD4 Effector")) # column_labels = c("Tpro1", "Tpro2", "Teff", "Texh")
dev.off()


ht = draw(h)
ht = row_order(ht) # a list of two 

for(i in 1:length(ht)){
  print(i)
  print(rownames(averaged_RNA)[ ht[[i]] ]) # the gene ids in the first cluster

}


```